// @university-of-missouri/mizzou-design-system/dist/es5/stories/components/Hero/Hero.stories.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Brand = exports.Default = void 0;
var jsx_runtime_1 = require("react/jsx-runtime");
var Hero_1 = __importDefault(require("Components/Hero/Hero"));
var config_json_1 = require("Components/Hero/config.json");
var Hero_mdx_1 = __importDefault(require("./Hero.mdx"));
var Template = function (args) {
    return (0, jsx_runtime_1.jsx)(Hero_1.default, __assign({}, args));
};
exports.default = {
    component: Hero_1.default,
    title: 'Design System/Components/Hero',
    parameters: {
        docs: {
            description: {
                component: config_json_1.description,
            },
            page: Hero_mdx_1.default,
        },
        status: config_json_1.status,
    },
    argTypes: {
        attributes: {
            table: {
                disable: true,
            },
        },
        title: {
            table: {
                category: 'Options',
            },
        },
        overline: {
            table: {
                category: 'Options',
            },
        },
        image: {
            table: {
                category: 'Options',
            },
        },
        buttons: {
            table: {
                category: 'Options',
            },
        },
        brand: {
            table: {
                category: 'Options',
            },
        },
        align: {
            table: {
                category: 'Options',
            },
        },
    },
};
exports.Default = Template.bind({});
exports.Default.args = {
    title: 'We’ll Show You How.',
    overline: 'Ready to Change the World?',
    image: '',
    buttons: [{ title: 'Apply Today' }, { title: 'Visit Campus' }],
    brand: false,
    align: 'left',
};
exports.Brand = Template.bind({});
exports.Brand.args = {
    title: 'We’ll Show You How.',
    overline: 'Ready to Change the World?',
    image: 'https://missouri.edu/sites/default/files/2020-07/CampusBeautyAerialOnJesse.jpg',
    buttons: [{ title: 'Apply Today' }, { title: 'Visit Campus' }],
    brand: true,
    align: 'center',
};
