// @university-of-missouri/mizzou-design-system/src/components/PersonProfile/PersonProfile.tsx
import React from 'react'
import DrupalAttribute from 'drupal-attribute'
import classNames from 'classnames'
import Card, {
  CardBody,
  CardSubtitle,
  CardTitle,
  CardText,
  CardLinks,
  CardImage,
  CardProps,
} from 'Components/Card/Card'

export interface PersonProfileProps extends CardProps {
  name: string | React.ReactNode
  position: string | React.ReactNode
  image?: any
  children: React.ReactNode
  links?: any
}

const PersonProfile = ({
  attributes = new DrupalAttribute(''),
  name = 'Truman Tiger',
  position = 'Mascot',
  children,
  links,
  image,
  theme = 'default',
  variant = 'default',
  branded = false,
}: PersonProfileProps): JSX.Element => {
  const CardAttributes = {
    className: classNames('miz-card--person-profile', attributes.get('class')),
  }

  attributes.forEach((value, key) => {
    if (key !== 'class') {
      CardAttributes[key] = value
    }
  })

  return (
    <Card
      variant={variant}
      branded={branded}
      theme={theme}
      attributes={attributes.addClass(CardAttributes.className)}
    >
      {image && <CardImage src={image.src} alt={image.alt} />}

      <CardBody>
        <CardTitle>{name}</CardTitle>
        {position && <CardSubtitle>{position}</CardSubtitle>}
        {children}
        {links && <CardLinks links={links} />}
      </CardBody>
    </Card>
  )
}

export default PersonProfile
