// @wordpress/block-library/build/navigation/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SELECT_NAVIGATION_MENUS_ARGS = exports.PRIORITIZED_INSERTER_BLOCKS = exports.PRELOADED_NAVIGATION_MENUS_QUERY = exports.DEFAULT_BLOCK = exports.ALLOWED_BLOCKS = void 0;
const DEFAULT_BLOCK = {
  name: 'core/navigation-link'
};
exports.DEFAULT_BLOCK = DEFAULT_BLOCK;
const ALLOWED_BLOCKS = ['core/navigation-link', 'core/search', 'core/social-links', 'core/page-list', 'core/spacer', 'core/home-link', 'core/site-title', 'core/site-logo', 'core/navigation-submenu', 'core/loginout'];
exports.ALLOWED_BLOCKS = ALLOWED_BLOCKS;
const PRIORITIZED_INSERTER_BLOCKS = ['core/navigation-link/page', 'core/navigation-link'];

// These parameters must be kept aligned with those in
// lib/compat/wordpress-6.3/navigation-block-preloading.php
// and
// edit-site/src/components/sidebar-navigation-screen-navigation-menus/constants.js
exports.PRIORITIZED_INSERTER_BLOCKS = PRIORITIZED_INSERTER_BLOCKS;
const PRELOADED_NAVIGATION_MENUS_QUERY = {
  per_page: 100,
  status: ['publish', 'draft'],
  order: 'desc',
  orderby: 'date'
};
exports.PRELOADED_NAVIGATION_MENUS_QUERY = PRELOADED_NAVIGATION_MENUS_QUERY;
const SELECT_NAVIGATION_MENUS_ARGS = ['postType', 'wp_navigation', PRELOADED_NAVIGATION_MENUS_QUERY];
exports.SELECT_NAVIGATION_MENUS_ARGS = SELECT_NAVIGATION_MENUS_ARGS;
