// @university-of-missouri/mizzou-design-system/dist/es5/stories/components/Alert/Alert.stories.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.alert = void 0;
var jsx_runtime_1 = require("react/jsx-runtime");
var Alert_1 = __importDefault(require("Components/Alert/Alert"));
var config_json_1 = require("Components/Alert/config.json");
var Alert_mdx_1 = __importDefault(require("./Alert.mdx"));
var Template = function (args) {
    return (0, jsx_runtime_1.jsx)(Alert_1.default, __assign({}, args));
};
exports.default = {
    component: Alert_1.default,
    title: 'Design System/Components/Alert',
    parameters: {
        docs: {
            description: {
                component: config_json_1.description,
            },
            page: Alert_mdx_1.default,
        },
        status: config_json_1.status,
    },
    argTypes: {
        children: {
            table: {
                category: 'Props',
            },
        },
        attributes: {
            table: {
                disable: true,
            },
        },
        link: {
            table: {
                category: 'Options',
            },
        },
        variant: {
            table: {
                category: 'Options',
            },
        },
    },
};
exports.alert = Template.bind({});
exports.alert.args = {
    children: (0, jsx_runtime_1.jsx)("p", { children: "Alert Content" }),
    variant: 'info',
    link: '',
};
