// @university-of-missouri/mizzou-design-system/dist/es5/components/Card/Card.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CardFooter = exports.CardBody = exports.CardHeader = exports.CardLinks = exports.CardLink = exports.CardText = exports.CardSubtitle = exports.CardTitle = exports.CardImage = void 0;
var jsx_runtime_1 = require("react/jsx-runtime");
var drupal_attribute_1 = __importDefault(require("drupal-attribute"));
var classnames_1 = __importDefault(require("classnames"));
var CardImage = function (_a) {
    var _b = _a.attributes, attributes = _b === void 0 ? new drupal_attribute_1.default('') : _b, src = _a.src, alt = _a.alt;
    var CardImageAttributes = {
        className: (0, classnames_1.default)('miz-card__image', attributes.get('class')),
    };
    attributes.forEach(function (value, key) {
        if (key !== 'class') {
            CardImageAttributes[key] = value;
        }
    });
    return (0, jsx_runtime_1.jsx)("img", __assign({}, CardImageAttributes, { src: src, alt: alt }));
};
exports.CardImage = CardImage;
var CardTitle = function (_a) {
    var _b = _a.attributes, attributes = _b === void 0 ? new drupal_attribute_1.default('') : _b, _c = _a.children, children = _c === void 0 ? 'Card Title' : _c;
    var CardTitleAttributes = {
        className: (0, classnames_1.default)('miz-card__title', attributes.get('class')),
    };
    attributes.forEach(function (value, key) {
        if (key !== 'class') {
            CardTitleAttributes[key] = value;
        }
    });
    return (0, jsx_runtime_1.jsx)("h2", __assign({}, CardTitleAttributes, { children: children }));
};
exports.CardTitle = CardTitle;
var CardSubtitle = function (_a) {
    var _b = _a.attributes, attributes = _b === void 0 ? new drupal_attribute_1.default('') : _b, children = _a.children;
    var CardSubtitleAttributes = {
        className: (0, classnames_1.default)('miz-card__subtitle', attributes.get('class')),
    };
    attributes.forEach(function (value, key) {
        if (key !== 'class') {
            CardSubtitleAttributes[key] = value;
        }
    });
    return (0, jsx_runtime_1.jsx)("h3", __assign({}, CardSubtitleAttributes, { children: children }));
};
exports.CardSubtitle = CardSubtitle;
var CardText = function (_a) {
    var _b = _a.attributes, attributes = _b === void 0 ? new drupal_attribute_1.default('') : _b, children = _a.children;
    var CardTextAttributes = {
        className: (0, classnames_1.default)('miz-card__text', attributes.get('class')),
    };
    attributes.forEach(function (value, key) {
        if (key !== 'class') {
            CardTextAttributes[key] = value;
        }
    });
    return (0, jsx_runtime_1.jsx)("p", __assign({}, CardTextAttributes, { children: children }));
};
exports.CardText = CardText;
var CardLink = function (_a) {
    var _b = _a.attributes, attributes = _b === void 0 ? new drupal_attribute_1.default('') : _b, children = _a.children, href = _a.href;
    var CardLinkAttributes = {
        className: (0, classnames_1.default)('miz-card__link', attributes.get('class')),
        href: href,
    };
    attributes.forEach(function (value, key) {
        if (key !== 'class') {
            CardLinkAttributes[key] = value;
        }
    });
    return (0, jsx_runtime_1.jsx)("a", __assign({}, CardLinkAttributes, { children: children }));
};
exports.CardLink = CardLink;
var CardLinks = function (_a) {
    var _b = _a.attributes, attributes = _b === void 0 ? new drupal_attribute_1.default('') : _b, links = _a.links;
    var CardLinksAttributes = {
        className: (0, classnames_1.default)('miz-card__links', attributes.get('class')),
    };
    attributes.forEach(function (value, key) {
        if (key !== 'class') {
            CardLinksAttributes[key] = value;
        }
    });
    var linksGroup;
    if (links) {
        linksGroup = links.map(function (link, i) { return ((0, jsx_runtime_1.jsx)(exports.CardLink, __assign({ href: link.href }, { children: link.text }), i)); });
    }
    return (0, jsx_runtime_1.jsx)("div", __assign({}, CardLinksAttributes, { children: linksGroup }));
};
exports.CardLinks = CardLinks;
var CardHeader = function (_a) {
    var _b = _a.attributes, attributes = _b === void 0 ? new drupal_attribute_1.default('') : _b, children = _a.children;
    var CardHeaderAttributes = {
        className: (0, classnames_1.default)('miz-card__header', attributes.get('class')),
    };
    attributes.forEach(function (value, key) {
        if (key !== 'class') {
            CardHeaderAttributes[key] = value;
        }
    });
    return (0, jsx_runtime_1.jsx)("div", __assign({}, CardHeaderAttributes, { children: children }));
};
exports.CardHeader = CardHeader;
var CardBody = function (_a) {
    var _b = _a.attributes, attributes = _b === void 0 ? new drupal_attribute_1.default('') : _b, children = _a.children;
    var CardBodyAttributes = {
        className: (0, classnames_1.default)('miz-card__body', attributes.get('class')),
    };
    attributes.forEach(function (value, key) {
        if (key !== 'class') {
            CardBodyAttributes[key] = value;
        }
    });
    return (0, jsx_runtime_1.jsx)("div", __assign({}, CardBodyAttributes, { children: children }));
};
exports.CardBody = CardBody;
var CardFooter = function (_a) {
    var _b = _a.attributes, attributes = _b === void 0 ? new drupal_attribute_1.default('') : _b, children = _a.children;
    var CardFooterAttributes = {
        className: (0, classnames_1.default)('miz-card__footer', attributes.get('class')),
    };
    attributes.forEach(function (value, key) {
        if (key !== 'class') {
            CardFooterAttributes[key] = value;
        }
    });
    return (0, jsx_runtime_1.jsx)("div", __assign({}, CardFooterAttributes, { children: children }));
};
exports.CardFooter = CardFooter;
var Card = function (_a) {
    var _b;
    var _c = _a.attributes, attributes = _c === void 0 ? new drupal_attribute_1.default('') : _c, children = _a.children, _d = _a.theme, theme = _d === void 0 ? 'default' : _d, _e = _a.variant, variant = _e === void 0 ? 'default' : _e, _f = _a.branded, branded = _f === void 0 ? false : _f;
    var CardAttributes = {
        className: (0, classnames_1.default)('miz-card', (_b = {
                'miz-card--dark': theme === 'dark'
            },
            _b["miz-card--".concat(variant)] = variant !== 'default',
            _b['miz-card--brand'] = branded,
            _b), attributes.get('class')),
    };
    attributes.forEach(function (value, key) {
        if (key !== 'class') {
            CardAttributes[key] = value;
        }
    });
    return (0, jsx_runtime_1.jsx)("div", __assign({}, CardAttributes, { children: children }));
};
exports.default = Card;
