// @university-of-missouri/mizzou-design-system/dist/es5/stories/components/SocialMedia/SocialMedia.stories.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.socialMedia = void 0;
var jsx_runtime_1 = require("react/jsx-runtime");
var SocialMedia_1 = __importDefault(require("Components/SocialMedia/SocialMedia"));
var config_json_1 = require("Components/SocialMedia/config.json");
var SocialMedia_mdx_1 = __importDefault(require("./SocialMedia.mdx"));
var Template = function (args) {
    return (0, jsx_runtime_1.jsx)(SocialMedia_1.default, __assign({}, args));
};
exports.default = {
    component: SocialMedia_1.default,
    title: 'Design System/Components/Social Media',
    parameters: {
        docs: {
            description: {
                component: config_json_1.description,
            },
            page: SocialMedia_mdx_1.default,
        },
        status: config_json_1.status,
    },
    argTypes: {
        attributes: {
            table: {
                disable: true,
            },
        },
    },
};
exports.socialMedia = Template.bind({});
exports.socialMedia.args = {
    size: 'md',
    color: 'default',
    name: 'twitter',
};
