// @university-of-missouri/mizzou-design-system/dist/es5/stories/components/Layers/BigMessage/BigMessage.stories.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.bigMessage = void 0;
var jsx_runtime_1 = require("react/jsx-runtime");
var BigMessage_1 = __importDefault(require("Components/Layers/BigMessage/BigMessage"));
var config_json_1 = require("Components/Layers/BigMessage/config.json");
var BigMessage_mdx_1 = __importDefault(require("./BigMessage.mdx"));
var Template = function (args) {
    return (0, jsx_runtime_1.jsx)(BigMessage_1.default, __assign({}, args));
};
exports.default = {
    component: BigMessage_1.default,
    title: 'Design System/Layers/Big Message',
    parameters: {
        docs: {
            description: {
                component: config_json_1.description,
            },
            page: BigMessage_mdx_1.default,
        },
        status: config_json_1.status,
    },
    argTypes: {
        title: {
            table: {
                category: 'Props',
            },
        },
        children: {
            table: {
                category: 'Props',
            },
        },
        attributes: {
            table: {
                disable: true,
            },
        },
        buttons: {
            table: {
                category: 'Options',
            },
        },
    },
};
exports.bigMessage = Template.bind({});
exports.bigMessage.args = {
    title: 'Big Message Title',
    children: 'Big Message Content',
    buttons: [{ title: 'Big Message Button', link: 'https://missouri.edu/' }],
};
