// @university-of-missouri/mizzou-design-system/dist/es5/stories/components/Layers/ByTheNumbers/ByTheNumbers.stories.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.byTheNumbers = void 0;
var jsx_runtime_1 = require("react/jsx-runtime");
var ByTheNumbers_1 = __importDefault(require("Components/Layers/ByTheNumbers/ByTheNumbers"));
var config_json_1 = require("Components/Layers/ByTheNumbers/config.json");
var ByTheNumbers_mdx_1 = __importDefault(require("./ByTheNumbers.mdx"));
var Template = function (args) {
    return (0, jsx_runtime_1.jsx)(ByTheNumbers_1.default, __assign({}, args));
};
exports.default = {
    component: ByTheNumbers_1.default,
    title: 'Design System/Layers/By The Numbers',
    parameters: {
        docs: {
            description: {
                component: config_json_1.description,
            },
            page: ByTheNumbers_mdx_1.default,
        },
        status: config_json_1.status,
    },
    argTypes: {
        title: {
            table: {
                category: 'Props',
            },
        },
        kicker: {
            table: {
                category: 'Props',
            },
        },
        attributes: {
            table: {
                disable: true,
            },
        },
        numbers: {
            table: {
                category: 'Options',
            },
        },
        buttons: {
            table: {
                category: 'Options',
            },
        },
    },
};
exports.byTheNumbers = Template.bind({});
exports.byTheNumbers.args = {
    title: 'By the Numbers Title',
    kicker: 'By the Numbers Kicker',
    numbers: [
        { number: '1', message: 'Number' },
        { number: '2', message: 'Number' },
        { number: '3', message: 'Number' },
    ],
    buttons: [{ title: 'By the Number Button', link: 'https://missouri.edu/' }],
};
