// @university-of-missouri/mizzou-design-system/src/stories/components/PageHeaders/PageHeader.stories.tsx
import React from 'react'
import { Story } from '@storybook/react'

import PageHeader, { PageHeaderProps } from 'Components/PageHeader/PageHeader'
import { description, status } from 'Components/PageHeader/config.json'
import mdx from './PageHeader.mdx'

const Template: Story<PageHeaderProps> = (args) => {
  return <PageHeader {...args} />
}

export default {
  component: PageHeader,
  title: 'Design System/Components/Page Headers',
  parameters: {
    docs: {
      description: {
        component: description,
      },
      page: mdx,
    },
    status,
  },
  argTypes: {
    title: {
      table: {
        category: 'Props',
      },
    },
    attributes: {
      table: {
        disable: true,
      },
    },
    branded: {
      table: {
        category: 'Options',
      },
    },
    variant: {
      table: {
        category: 'Options',
      },
      options: ['default', 'dark'],
      control: {
        type: 'radio',
      },
    },
    size: {
      table: {
        category: 'Options',
      },
      options: ['default', 'small', 'large', 'xlarge'],
      control: {
        type: 'select',
      },
    },
  },
}

export const pageHeaders = Template.bind({})
pageHeaders.args = {
  title: 'Base Page Header',
  branded: false,
  variant: 'default',
  size: 'default',
}
