// @university-of-missouri/mizzou-design-system/src/stories/components/Layers/CallToAction/CallToAction.stories.tsx
import React from 'react'
import { Story } from '@storybook/react'

import CallToAction, { CallToActionProps } from 'Components/Layers/CallToAction/CallToAction'
import { description, status } from 'Components/Layers/CallToAction/config.json'
import mdx from './CallToAction.mdx'

const Template: Story<CallToActionProps> = (args) => {
  return <CallToAction {...args} />
}

export default {
  component: CallToAction,
  title: 'Design System/Layers/Call To Action',
  parameters: {
    docs: {
      description: {
        component: description,
      },
      page: mdx,
    },
    status,
  },
  argTypes: {
    title: {
      table: {
        category: 'Props',
      },
    },
    children: {
      table: {
        category: 'Props',
      },
    },
    attributes: {
      table: {
        disable: true,
      },
    },
    variant: {
      table: {
        category: 'Options',
      },
    },
    image: {
      table: {
        category: 'Options',
      },
    },
    buttons: {
      table: {
        category: 'Options',
      },
    },
  },
}
export const callToAction = Template.bind({})
callToAction.args = {
  title: 'Call to Action Title',
  children: 'Call to Action Content',
  variant: 'black',
  image: 'https://missouri.edu/sites/default/files/2020-07/CampusBeautyAerialOnJesse.jpg',
  buttons: [{ title: 'Call to Action Button', link: 'https://missouri.edu/' }],
}
