// @university-of-missouri/mizzou-design-system/src/stories/components/Layers/BigMessage/BigMessage.stories.tsx
import React from 'react'
import { Story } from '@storybook/react'

import BigMessage, { BigMessageProps } from 'Components/Layers/BigMessage/BigMessage'
import { description, status } from 'Components/Layers/BigMessage/config.json'
import mdx from './BigMessage.mdx'

const Template: Story<BigMessageProps> = (args) => {
  return <BigMessage {...args} />
}

export default {
  component: BigMessage,
  title: 'Design System/Layers/Big Message',
  parameters: {
    docs: {
      description: {
        component: description,
      },
      page: mdx,
    },
    status,
  },
  argTypes: {
    title: {
      table: {
        category: 'Props',
      },
    },
    children: {
      table: {
        category: 'Props',
      },
    },
    attributes: {
      table: {
        disable: true,
      },
    },
    buttons: {
      table: {
        category: 'Options',
      },
    },
  },
}
export const bigMessage = Template.bind({})
bigMessage.args = {
  title: 'Big Message Title',
  children: 'Big Message Content',
  buttons: [{ title: 'Big Message Button', link: 'https://missouri.edu/' }],
}
