// @university-of-missouri/mizzou-design-system/dist/es5/stories/components/Layers/LinkList/LinkListLayer.stories.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.linkList = void 0;
var jsx_runtime_1 = require("react/jsx-runtime");
var LinkList_1 = __importDefault(require("Components/Layers/LinkList/LinkList"));
var config_json_1 = require("Components/Layers/LinkList/config.json");
var LinkList_mdx_1 = __importDefault(require("./LinkList.mdx"));
var Template = function (args) {
    return (0, jsx_runtime_1.jsx)(LinkList_1.default, __assign({}, args));
};
exports.default = {
    component: LinkList_1.default,
    title: 'Design System/Layers/Link List',
    parameters: {
        docs: {
            description: {
                component: config_json_1.description,
            },
            page: LinkList_mdx_1.default,
        },
        status: config_json_1.status,
    },
    argTypes: {
        attributes: {
            table: {
                disable: true,
            },
        },
        layout: {
            table: {
                category: 'Options',
                disable: true,
            },
        },
        align: {
            table: {
                category: 'Options',
                disable: true,
            },
        },
        fill: {
            table: {
                category: 'Options',
                disable: true,
            },
        },
        branded: {
            table: {
                category: 'Options',
                disable: true,
            },
        },
        list: {
            table: {
                category: 'Options',
            },
        },
    },
};
exports.linkList = Template.bind({});
exports.linkList.args = {
    title: 'Link List Title',
    list: [
        { text: 'List Item 1', url: '#' },
        { text: 'List Item 2', url: '#' },
        { text: 'List Item 3', url: '#' },
    ],
};
