// @wordpress/block-library/build/page-list/convert-to-links-modal.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ConvertToLinksModal = ConvertToLinksModal;
exports.convertDescription = void 0;
var _element = require("@wordpress/element");
var _components = require("@wordpress/components");
var _i18n = require("@wordpress/i18n");
/**
 * WordPress dependencies
 */

const convertDescription = (0, _i18n.__)('This page list is synced with the published pages on your site. Detach the page list to add, delete, or reorder pages yourself.');
exports.convertDescription = convertDescription;
function ConvertToLinksModal({
  onClick,
  onClose,
  disabled
}) {
  return (0, _element.createElement)(_components.Modal, {
    onRequestClose: onClose,
    title: (0, _i18n.__)('Edit Page List'),
    className: 'wp-block-page-list-modal',
    aria: {
      describedby: 'wp-block-page-list-modal__description'
    }
  }, (0, _element.createElement)("p", {
    id: 'wp-block-page-list-modal__description'
  }, convertDescription), (0, _element.createElement)("div", {
    className: "wp-block-page-list-modal-buttons"
  }, (0, _element.createElement)(_components.Button, {
    variant: "tertiary",
    onClick: onClose
  }, (0, _i18n.__)('Cancel')), (0, _element.createElement)(_components.Button, {
    variant: "primary",
    disabled: disabled,
    onClick: onClick
  }, (0, _i18n.__)('Detach'))));
}
