// @university-of-missouri/mizzou-design-system/dist/es5/stories/patterns/Pagination/Pagination.stories.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.pagination = void 0;
var jsx_runtime_1 = require("react/jsx-runtime");
var Pagination_1 = __importDefault(require("Components/Pagination/Pagination"));
var config_json_1 = require("Components/Pagination/config.json");
var Pagination_mdx_1 = __importDefault(require("./Pagination.mdx"));
var StoryTemplate = function (args, _a) {
    var fileTypes = _a.globals.fileTypes;
    if (fileTypes === 'react') {
        return (0, jsx_runtime_1.jsx)(Pagination_1.default, __assign({}, args));
    }
    return (0, jsx_runtime_1.jsx)("div", { children: "Null" });
};
exports.default = {
    component: Pagination_1.default,
    title: 'Design System/Patterns/Pagination',
    parameters: {
        docs: {
            description: {
                component: config_json_1.description,
            },
            page: Pagination_mdx_1.default,
        },
        status: config_json_1.status,
    },
    argTypes: {
        children: {
            table: {
                category: 'Props',
            },
        },
        attributes: {
            table: {
                disable: true,
            },
        },
        branded: {
            table: {
                category: 'Options',
            },
        },
    },
};
exports.pagination = StoryTemplate.bind({});
exports.pagination.args = {};
