// @university-of-missouri/mizzou-design-system/src/stories/components/Layers/FeaturedNews/FeaturedNews.stories.tsx
import React from 'react'
import { Story } from '@storybook/react'

import FeaturedNews, { FeaturedNewsProps } from 'Components/Layers/FeaturedNews/FeaturedNews'
import { description, status } from 'Components/Layers/FeaturedNews/config.json'
import mdx from './FeaturedNews.mdx'

const Template: Story<FeaturedNewsProps> = (args) => {
  return <FeaturedNews {...args} />
}

export default {
  component: FeaturedNews,
  title: 'Design System/Layers/Featured News',
  parameters: {
    docs: {
      description: {
        component: description,
      },
      page: mdx,
    },
    status,
  },
  argTypes: {
    title: {
      table: {
        category: 'Options',
      },
    },
    kicker: {
      table: {
        category: 'Options',
      },
    },
    posts: {
      table: {
        category: 'Options',
      },
    },
    buttons: {
      table: {
        category: 'Options',
      },
    },
  },
}
export const featuredNews = Template.bind({})
featuredNews.args = {
  title: 'Featured News',
  kicker: 'Mizzou Made',
  buttons: [{ title: 'More News', link: 'https://missouri.edu/' }],
  posts: [
    {
      title: 'Mizzou on the moon',
      content:
        // eslint-disable-next-line prettier/prettier
        '50th anniversary of the moon landing shines spotlight on MU\'s ties to the space program.',
      image: 'https://missouri.edu/sites/default/files/2020-07/CampusBeautyAerialOnJesse.jpg',
      branded: false,
    },
    {
      title: 'Mizzou on the moon',
      content:
        // eslint-disable-next-line prettier/prettier
        '50th anniversary of the moon landing shines spotlight on MU\'s ties to the space program.',
      image: 'https://missouri.edu/sites/default/files/2020-07/CampusBeautyAerialOnJesse.jpg',
      branded: false,
    },
    {
      title: 'Mizzou on the moon',
      content:
        // eslint-disable-next-line prettier/prettier
        '50th anniversary of the moon landing shines spotlight on MU\'s ties to the space program.',
      image: 'https://missouri.edu/sites/default/files/2020-07/CampusBeautyAerialOnJesse.jpg',
      branded: false,
    },
  ],
}
