// @university-of-missouri/mizzou-design-system/.storybook/main.js
const mizWebpackDev = require('./../webpack/dev.config')

module.exports = {
  core: {
    builder: 'webpack5',
  },
  stories: [
    '../src/stories/**/*.stories.mdx',
    '../src/stories/**/*.stories.@(js|jsx|ts|tsx)'
  ],
  addons: [
    '@storybook/addon-links',
    '@storybook/addon-essentials',
    '@storybook/addon-a11y',
    '@whitespace/storybook-addon-html',
  ],
  features: {
    buildStoriesJson: true,
  },
  staticDirs: [{ from: '../src/assets', to: '/' }],
  webpackFinal: (config) => {
    config.entry = config.entry.filter(singleEntry => !singleEntry.includes('/webpack-hot-middleware/'))
    return [{
      ...config,
      resolve: {
        ...config.resolve,
        alias: {
          ...config.resolve.alias,
          ...mizWebpackDev[0].resolve.alias
        },
      },
      module: {
        ...config.module,
        rules: config.module.rules.concat(mizWebpackDev[0].module.rules)
      },
      plugins: config.plugins.concat(mizWebpackDev[0].plugins),
    }, mizWebpackDev[1]]
  },
}
