// @university-of-missouri/mizzou-design-system/.storybook/preview.js
import React from 'react'
import '@storybook/addon-console'
import { DocsContainer } from "@storybook/addon-docs"

import './style.scss'
import './../src/scss/miz.scss'
import './../src/scss/miz-brand.scss'
import './../src/scss/miz-fonts.scss'

export const globalTypes = {
  fileTypes: {
    name: 'File type',
    description: 'Twig or React',
    defaultValue: 'react',
    toolbar: {
      icon: 'markup',
      items: [
        {value: 'react', title: 'React'},
        {value: 'twig', title: 'Twig'},
      ],
      showName: false,
    }
  },
  docTypes: {
    name: 'Documentation Type',
    description: 'View documentation for content editor or developer.',
    defaultValue: 'editor',
    toolbar: {
      icon: 'user',
      items: [
        {value: 'editor', title: 'Content Editor'},
        {value: 'dev', title: 'Developer'},
      ],
      showName: false,
    }
  }
}

export const parameters = {
  actions: { argTypesRegex: "^on[A-Z].*" },
  backgrounds: {
    default: 'light',
    values: [
      {
        name: 'light',
        value: '#FFF',
      },
      {
        name: 'dark',
        value: '#333',
      }
    ],
  },
  controls: {
    matchers: {
      color: /(background|color)$/i,
      date: /Date$/,
    },
  },
  docs: {
    container: ({ children, context }) => <DocsContainer context={context}><div className="miz-main">{children}</div></DocsContainer>,
    components: {
      p: ({ children, className }) => <p className={className}>{children}</p>,
      h1: ({ children, className }) => <h1 className={className}>{children}</h1>,
      h2: ({ children, className }) => <h2 className={className}>{children}</h2>,
      h3: ({ children, className }) => <h3 className={className}>{children}</h3>,
      h4: ({ children, className }) => <h4 className={className}>{children}</h4>,
      h5: ({ children, className }) => <h5 className={className}>{children}</h5>,
      h6: ({ children, className }) => <h6 className={className}>{children}</h6>,
      a: ({ children, className, href }) => <a href={href} className={className}>{children}</a>,
      ul: ({ children, className }) => <ul className={className}>{children}</ul>,
      li: ({ children, className }) => <li className={className}>{children}</li>,
    },
  },
  layout: 'padded',
  options: {
    storySort: {
      order: ['Overview', 'Documentation', 'Layout', 'Style', 'Components', 'Layers', 'Templates']
    }
  },
  previewTabs: {
    'storybook/docs/panel': {
      index: -1,
      title: 'Documentation',
    },
    canvas: { hidden: false },
  },
  viewMode: 'docs',
}
