// @university-of-missouri/mizzou-design-system/src/stories/components/Navigation/MegaMenu/MegaMenu.context.js
module.exports = {
  default: {
    nav: {
      class: 'miz-nav',
      list: {
        class: 'miz-nav__list',
        items: [
          {
            content: {
              class: 'miz-nav__list-item',
              link: { class: 'miz-nav__link', href: '/', text: 'Home' },
            },
          },
          {
            content: {
              class: 'miz-nav__list-item',
              link: { class: 'miz-nav__link', href: '/about-us', text: 'About us' },
            },
          },
          {
            content: {
              class: 'miz-nav__list-item',
              link: { class: 'miz-nav__link', href: '/departments', text: 'Departments' },
            },
          },
          {
            content: {
              class: 'miz-nav__list-item',
              button: {
                class: 'miz-nav__link',
                type: 'button',
                content: 'Other',
              },
              list: {
                class: 'miz-nav__list miz-nav__sub-list',
                // id: 'subNavListId',
                items: [
                  {
                    // attributes: {
                    //   id: 'navItem1',
                    // },
                    content: {
                      class: 'miz-nav__list-item',
                      link: {
                        class: 'miz-nav__link',
                        href: 'https://mualert.missouri.edu/',
                        text: 'MU Alert',
                      },
                    },
                  },
                  {
                    // attributes: {
                    //   id: 'navItem2'
                    // },
                    content: {
                      class: 'miz-nav__list-item',
                      link: {
                        class: 'miz-nav__link',
                        href: 'https://missouri.edu/',
                        text: 'MU Homepage',
                      },
                    },
                  },
                ],
              },
            },
          },
        ],
      },
    },
  },
}
