// @university-of-missouri/mizzou-design-system/dist/es5/stories/components/Masthead/Masthead.stories.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.masthead = void 0;
var jsx_runtime_1 = require("react/jsx-runtime");
var Masthead_1 = __importDefault(require("Components/Masthead/Masthead"));
var config_json_1 = require("Components/Masthead/config.json");
var Masthead_mdx_1 = __importDefault(require("./Masthead.mdx"));
var Template = function (args) {
    return (0, jsx_runtime_1.jsx)(Masthead_1.default, __assign({}, args));
};
exports.default = {
    component: Masthead_1.default,
    title: 'Design System/Components/Masthead',
    parameters: {
        docs: {
            description: {
                component: config_json_1.description,
            },
            page: Masthead_mdx_1.default,
        },
        status: config_json_1.status,
    },
    argTypes: {
        title: {
            table: {
                category: 'Props',
            },
        },
        subtitle: {
            table: {
                category: 'Props',
            },
        },
        attributes: {
            table: {
                disable: true,
            },
        },
    },
};
exports.masthead = Template.bind({});
exports.masthead.args = {
    title: 'Default Site Name',
    subtitle: '',
    branded: false,
};
