// @university-of-missouri/mizzou-design-system/src/components/Masthead/Masthead.tsx
import * as React from 'react'
import DrupalAttribute from 'drupal-attribute'
import classNames from 'classnames'

export interface MastheadProps {
  attributes?: DrupalAttribute
  title?: string
  subtitle?: string
  link?: string
  branded?: boolean
  variant?: 'default' | 'dark'
  size?: 'small' | 'medium' | 'large'
}

const Masthead = ({
  attributes = new DrupalAttribute(''),
  title = 'Default Site Name',
  subtitle,
  link = '/',
  branded = false,
  size = 'medium',
  variant = 'default',
}: MastheadProps): JSX.Element => {
  const MastheadAttributes = {
    className: classNames(
      'miz-masthead',
      {
        'miz-masthead--brand': branded,
        [`miz-masthead--${size}`]: size !== 'medium',
        'miz-masthead--dark': variant === 'dark',
      },
      attributes.get('class'),
    ),
  }

  attributes.forEach((value, key) => {
    if (key !== 'class') {
      MastheadAttributes[key] = value
    }
  })

  return (
    <header {...MastheadAttributes} role="banner">
      <h1 className="miz-masthead__title">
        <a title={title} href={link}>
          {title}
        </a>
      </h1>
      {subtitle && <p className="miz-masthead__subtitle">{subtitle}</p>}
    </header>
  )
}

export default Masthead
