// @university-of-missouri/mizzou-design-system/dist/es5/stories/components/PersonProfile/PersonProfile.stories.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.personProfile = void 0;
var jsx_runtime_1 = require("react/jsx-runtime");
var Card_1 = require("Components/Card/Card");
var PersonProfile_1 = __importDefault(require("Components/PersonProfile/PersonProfile"));
var config_json_1 = require("Components/PersonProfile/config.json");
var PersonProfile_mdx_1 = __importDefault(require("./PersonProfile.mdx"));
var Template = function (args) {
    return (0, jsx_runtime_1.jsx)(PersonProfile_1.default, __assign({}, args));
};
exports.default = {
    component: PersonProfile_1.default,
    title: 'Design System/Components/Person Profile',
    parameters: {
        docs: {
            description: {
                component: config_json_1.description,
            },
            page: PersonProfile_mdx_1.default,
        },
        status: config_json_1.status,
    },
    argTypes: {
        name: {
            table: {
                category: 'Props',
            },
        },
        position: {
            table: {
                category: 'Props',
            },
        },
        children: {
            table: {
                category: 'Props',
            },
        },
        attributes: {
            table: {
                disable: true,
            },
        },
        theme: {
            options: ['default', 'dark'],
            table: {
                category: 'Options',
            },
        },
        branded: {
            table: {
                category: 'Options',
            },
        },
        image: {
            table: {
                category: 'Options',
            },
        },
        links: {
            table: {
                category: 'Options',
            },
        },
        variant: {
            table: {
                category: 'Options',
            },
        },
    },
};
exports.personProfile = Template.bind({});
exports.personProfile.args = {
    name: 'Truman Tiger',
    position: 'Mascot',
    children: (0, jsx_runtime_1.jsx)(Card_1.CardText, { children: "Truman was acclaimed the Best Mascot in the Nation." }),
    image: {
        src: 'https://showme.missouri.edu/wp-content/uploads/2021/09/091321Truman1-600x600.jpg',
        alt: 'Truman Tiger',
    },
    variant: 'inline',
    theme: 'default',
    branded: false,
};
