// @university-of-missouri/mizzou-design-system/webpack/common.config.js
const path = require('path')

const ESLintPlugin = require('eslint-webpack-plugin')

const config = {
  context: path.resolve(__dirname, '../src'),
  plugins: [
    new ESLintPlugin({
      context: path.resolve(__dirname, '../src'),
    }),
  ],
  module: {
    rules: [
      {
        test: /\.(js|jsx)$/,
        exclude: /node_modules/,
        use: {
          loader: 'babel-loader',
          options: {
            presets: ['@babel/preset-env'],
          },
        },
      },
      {
        test: /\.(ts|tsx)$/,
        use: [
          {
            loader: 'ts-loader',
            options: {
              compilerOptions: {
                rootDir: null,
              },
            },
          },
        ],
        exclude: /node_modules/,
      },
    ],
  },
  resolve: {
    alias: {
      Components: path.resolve(__dirname, '../src/components/'),
      Layers: path.resolve(__dirname, '../src/components/Layers/'),
      Layout: path.resolve(__dirname, '../src/components/Layout/'),
      SB: path.resolve(__dirname, '../.storybook/'),
    },
    extensions: ['.tsx', '.ts', '.jsx', '.js', '...'],
    modules: [path.resolve(__dirname, '../'), 'node_modules'],
  },
}

module.exports = config
