// @university-of-missouri/mizzou-design-system/webpack/prod.config.js
const { merge } = require('webpack-merge')
const sass = require('sass')
const MiniCssExtractPlugin = require('mini-css-extract-plugin')
const TwigModify = require('./plugins/TwigModify')

const baseConfig = require('./common.config')

const scriptsConfig = merge(baseConfig, {
  entry: {
    scripts: {
      import: './scripts.js',
      filename: 'scripts/scripts.js',
    },
    dropdown: {
      import: './scripts/dropdown.js',
    },
    expand: {
      import: './scripts/expand.js',
    },
    toggle: {
      import: './scripts/toggle.js',
    },
    utils: {
      import: './scripts/utils.js',
    },
    primaryNavigation: {
      import: './components/Navigation/Primary/primary.js',
      library: {
        name: 'primaryNavigation',
        type: 'umd',
      },
    },
  },
  output: {
    filename: 'scripts/[name]/[name].js',
    library: {
      name: 'mizScripts',
      type: 'umd',
    },
  },
})

const config = merge(baseConfig, {
  entry: {
    images: './images.js',
    styles: './styles.js',
    twig: './twig.js',
  },
  plugins: [
    new MiniCssExtractPlugin({
      filename: '[name].css',
    }),
    new TwigModify(),
  ],
  module: {
    rules: [
      {
        test: /\.(woff|woff2|eot|ttf|otf)$/i,
        type: 'asset/resource',
        generator: {
          filename: (pathData) => {
            const pathToFonts = 'src/assets/fonts/'
            const { context } = pathData.module
            const folder = context.substring(context.indexOf(pathToFonts) + pathToFonts.length)
            return `assets/fonts/${folder}/[name][ext]`
          },
        },
      },
      {
        test: /\.(png|jpg|gif|bmp|svg|ico)$/i,
        type: 'asset/resource',
        generator: {
          filename: (pathData) => {
            const pathToImages = 'src/assets/images/'
            const { context } = pathData.module
            const folder = context.substring(context.indexOf(pathToImages) + pathToImages.length)
            return `assets/images/${folder}/[name][ext]`
          },
        },
      },
      {
        test: /\.scss$/i,
        use: [
          MiniCssExtractPlugin.loader,
          { loader: 'css-loader', options: { sourceMap: true } },
          { loader: 'resolve-url-loader', options: { sourceMap: true } },
          {
            loader: 'postcss-loader',
            options: {
              sourceMap: true,
              postcssOptions: {
                plugins: [['postcss-preset-env']],
              },
            },
          },
          {
            loader: 'sass-loader',
            options: {
              sourceMap: true,
              implementation: sass,
              sassOptions: {
                includePaths: ['./src/scss', './src'],
              },
            },
          },
        ],
      },
      {
        test: /\.twig$/i,
        type: 'asset/resource',
        generator: {
          filename: (pathData) => {
            const { rawRequest } = pathData.module

            const rawPath = rawRequest.split('/')

            rawPath.forEach((pathSection, i) => {
              rawPath[i] = pathSection
                .replace(/(.)([A-Z][a-z]+)/, '$1-$2')
                .replace(/([a-z0-9])([A-Z])/, '$1-$2')
                .toLowerCase()
            })

            return `twig/${rawPath.join('/').toString()}`
          },
        },
      },
    ],
  },
  optimization: {
    removeEmptyChunks: true,
    splitChunks: {
      cacheGroups: {
        scss: {
          name: (module) => {
            const moduleFileName = module
              .identifier()
              .split('/')
              .reduceRight((item) => item)
              .replace(/\.[^/.]+$/, '')

            return `assets/css/${moduleFileName}`
          },
          type: 'css/mini-extract',
          chunks: 'all',
          enforce: true,
        },
      },
    },
  },
})

module.exports = [config, scriptsConfig]
