// @university-of-missouri/mizzou-design-system/.storybook/theme/base.ts
import MizzouLogo from './../../src/assets/images/signatures/university/mu-logo-48.svg'

export const branding = {
  title: 'Mizzou Design System Storybook',
  url: null,
  image: MizzouLogo,
}

export const colors = {
  // Base
  black: '#111111', // miz-black
  gold: '#F1B82D', // miz-gold
  white: '#FFFFFF', // miz-white

  // Black
  black100: '#F0F0F0',
  black200: '#CCCCCC',
  black300: '#999999',
  black400: '#666666',
  black500: '#444444',
  black600: '#222222',

  // Gold
  gold100: '#FFF4D6',
  gold200: '#FEE8B6',
  gold300: '#FBD986',
  gold400: '#F1B82D',
  gold500: '#D79900',
  gold600: '#AF7C00',

  // Blue
  blue: '#318FD7',
  blue100: '#E4F1FA',
  blue200: '#B2D5F0',
  blue300: '#65ABE1',
  blue400: '#318FD7',
  blue500: '#1A5888',
  blue600: '#143650',

  // Green
  green: '#3F8F26',
  green100: '#EBF4E9',
  green200: '#B2CFA9',
  green300: '#6FAB5D',
  green400: '#3F8F26',
  green500: '#2D671B',
  green600: '#214716',

  // Orange
  orange: '#BD4D1F',
  orange100: '#FFE4DB',
  orange200: '#FFBEA9',
  orange300: '#DE6835',
  orange400: '#BD4D1F',
  orange500: '#992D00',
  orange600: '#611C00',

  // Red
  red: '#900000',
  red100: '#FFECEC',
  red200: '#FFC2C4',
  red300: '#CA4B4B',
  red400: '#900000',
  red500: '#6A0206',
  red600: '#470102',
}

export const typography = {
  fonts: {
    base: [
      '"Open Sans"',
      "Segoe UI",
      'Tahoma',
      'sans-serif'
    ].join(', '),
    mono: [
      'Menlo',
      'Consolas',
      'monospace'
    ].join(', '),
  },
  weight: {
    regular: 400,
    light: 300,
    semibold: 600,
    bold: 700,
    bolder: 800,
  },
  sizes: {}
}
