// @university-of-missouri/mizzou-design-system/src/stories/components/Layers/ByTheNumbers/ByTheNumbers.stories.tsx
import React from 'react'
import { Story } from '@storybook/react'

import ByTheNumbers, { ByTheNumbersProps } from 'Components/Layers/ByTheNumbers/ByTheNumbers'
import { description, status } from 'Components/Layers/ByTheNumbers/config.json'
import mdx from './ByTheNumbers.mdx'

const Template: Story<ByTheNumbersProps> = (args) => {
  return <ByTheNumbers {...args} />
}

export default {
  component: ByTheNumbers,
  title: 'Design System/Layers/By The Numbers',
  parameters: {
    docs: {
      description: {
        component: description,
      },
      page: mdx,
    },
    status,
  },
  argTypes: {
    title: {
      table: {
        category: 'Props',
      },
    },
    kicker: {
      table: {
        category: 'Props',
      },
    },
    attributes: {
      table: {
        disable: true,
      },
    },
    numbers: {
      table: {
        category: 'Options',
      },
    },
    buttons: {
      table: {
        category: 'Options',
      },
    },
  },
}
export const byTheNumbers = Template.bind({})
byTheNumbers.args = {
  title: 'By the Numbers Title',
  kicker: 'By the Numbers Kicker',
  numbers: [
    { number: '1', message: 'Number' },
    { number: '2', message: 'Number' },
    { number: '3', message: 'Number' },
  ],
  buttons: [{ title: 'By the Number Button', link: 'https://missouri.edu/' }],
}
