// @university-of-missouri/mizzou-design-system/src/components/Layers/ByTheNumbers/ByTheNumbers.tsx
import React from 'react'
import { AnchorButton } from 'Components/Button/Button'
import classNames from 'classnames'

export interface ByTheNumbersProps {
  title: string | React.ReactNode
  kicker?: string | React.ReactNode
  numbers?: any
  buttons?: any
  variant?: 'default' | 'dark'
}

const ByTheNumbers = ({
  title = 'Big Message Title',
  kicker,
  numbers,
  buttons,
  variant = 'default',
}: ByTheNumbersProps): JSX.Element => {
  let numbersGroup
  let buttonGroup

  const sectionClass = classNames('miz-layer miz-layer--brand miz-main-grid__full', {
    'miz-by-the-numbers--dark': variant === 'dark',
  })

  if (numbers) {
    numbersGroup = numbers.map((number, i) => (
      <div className="miz-by-the-numbers__column" key={i}>
        <p className="miz-by-the-numbers__number miz-text--black">{number.number}</p>
        <p>{number.message}</p>
      </div>
    ))
  }

  if (buttons) {
    buttonGroup = buttons.map((button, i) => (
      <AnchorButton branded={true} primary={true} link={button.link} key={i}>
        {button.title}
      </AnchorButton>
    ))
  }

  return (
    <section className={sectionClass}>
      <div className="miz-container">
        <header className="miz-layer__header miz-layer__header--center">
          {kicker && <p className="miz-layer__kicker">{kicker}</p>}

          <h2 className="miz-layer__title">{title}</h2>
        </header>

        <div className="miz-layer__content miz-layer__content--center">
          {numbers && <div className="miz-by-the-numbers__row">{numbersGroup}</div>}

          {buttons && <div className="miz-button-group">{buttonGroup}</div>}
        </div>
      </div>
    </section>
  )
}

export default ByTheNumbers
