// @university-of-missouri/mizzou-design-system/dist/es5/stories/components/BlockMessage/BlockMessage.stories.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ThemedVariants = exports.Base = void 0;
var jsx_runtime_1 = require("react/jsx-runtime");
var BlockMessage_1 = __importDefault(require("Components/BlockMessage/BlockMessage"));
var config_json_1 = require("Components/BlockMessage/config.json");
var BlockMessage_mdx_1 = __importDefault(require("./BlockMessage.mdx"));
var Template = function (args) {
    return (0, jsx_runtime_1.jsx)(BlockMessage_1.default, __assign({}, args));
};
exports.default = {
    component: BlockMessage_1.default,
    title: 'Design System/Components/Block Message',
    parameters: {
        docs: {
            description: {
                component: config_json_1.description,
            },
            page: BlockMessage_mdx_1.default,
        },
        status: config_json_1.status,
    },
    argTypes: {
        title: {
            table: {
                category: 'Props',
            },
        },
        children: {
            table: {
                category: 'Props',
            },
        },
        attributes: {
            table: {
                disable: true,
            },
        },
        variant: {
            table: {
                category: 'Options',
            },
        },
    },
};
exports.Base = Template.bind({});
exports.Base.args = {
    title: 'Block Message',
    children: 'Message Content',
};
exports.ThemedVariants = Template.bind({});
exports.ThemedVariants.args = {
    title: 'Block Message Message',
    children: 'Block Message Content',
    variant: 'info',
};
