// @university-of-missouri/mizzou-design-system/dist/es5/stories/components/Navigation/Breadcrumbs.stories.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.breadcrumbs = void 0;
var jsx_runtime_1 = require("react/jsx-runtime");
var Breadcrumbs_1 = __importDefault(require("Components/Breadcrumbs/Breadcrumbs"));
var config_json_1 = require("Components/Breadcrumbs/config.json");
// import TwigBreadcrumbs from './Breadcrumbs.twig'
var Breadcrumbs_mdx_1 = __importDefault(require("./Breadcrumbs/Breadcrumbs.mdx"));
var Template = function (args) {
    return (0, jsx_runtime_1.jsx)(Breadcrumbs_1.default, __assign({}, args));
};
exports.default = {
    component: Breadcrumbs_1.default,
    title: 'Design System/Components/Navigation/Breadcrumbs',
    decorators: [
        function (BreadcrumbsStory) { return ((0, jsx_runtime_1.jsx)("div", __assign({ className: "miz-main-grid miz-fill--black-100" }, { children: (0, jsx_runtime_1.jsx)(BreadcrumbsStory, {}) }))); },
    ],
    parameters: {
        docs: {
            description: {
                component: config_json_1.description,
            },
            page: Breadcrumbs_mdx_1.default,
        },
        status: config_json_1.status,
    },
    argTypes: {
        children: {
            table: {
                category: 'Props',
            },
        },
        attributes: {
            table: {
                disable: true,
            },
        },
        branded: {
            table: {
                category: 'Options',
            },
        },
    },
};
exports.breadcrumbs = Template.bind({});
exports.breadcrumbs.args = {
    branded: false,
    list: {
        items: [
            {
                link: {
                    text: 'Home',
                    href: '/',
                },
            },
            {
                link: {
                    text: 'About us',
                    href: '/crumb-1/',
                },
            },
            {
                link: {
                    text: 'About us',
                    href: '/crumb-1/',
                },
            },
            {
                page: {
                    text: 'Page title',
                },
            },
        ],
    },
};
