// @university-of-missouri/mizzou-design-system/dist/es5/stories/components/Navigation/MegaMenu.stories.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.megaMenu = void 0;
var jsx_runtime_1 = require("react/jsx-runtime");
var react_1 = __importDefault(require("react"));
var MegaMenu_1 = __importDefault(require("Components/MegaMenu/MegaMenu"));
var MegaMenu_twig_1 = __importDefault(require("Components/MegaMenu/MegaMenu.twig"));
var config_json_1 = require("Components/MegaMenu/config.json");
var Nav_1 = __importDefault(require("Components/Nav/Nav"));
var dropdown_1 = __importDefault(require("../../../scripts/dropdown"));
var MegaMenu_mdx_1 = __importDefault(require("./MegaMenu/MegaMenu.mdx"));
var MegaMenu_context_1 = __importDefault(require("./MegaMenu/MegaMenu.context"));
var Template = function (args, _a) {
    var fileTypes = _a.globals.fileTypes;
    react_1.default.useEffect(function () {
        (0, dropdown_1.default)();
    }, []);
    if (fileTypes === 'react') {
        return (0, jsx_runtime_1.jsx)(MegaMenu_1.default, __assign({}, args));
    }
    return (0, jsx_runtime_1.jsx)("div", { dangerouslySetInnerHTML: { __html: (0, MegaMenu_twig_1.default)() } });
};
exports.default = {
    component: MegaMenu_1.default,
    title: 'Design System/Components/Navigation/Mega Menu',
    parameters: {
        docs: {
            description: {
                component: config_json_1.description,
            },
            page: MegaMenu_mdx_1.default,
        },
        status: config_json_1.status,
    },
    argTypes: {},
    subcomponents: { Nav: Nav_1.default },
};
exports.megaMenu = Template.bind({});
exports.megaMenu.args = __assign({}, MegaMenu_context_1.default.default);
