// @university-of-missouri/mizzou-design-system/dist/es5/stories/components/Navigation/Primary.stories.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.primaryNavigation = void 0;
var jsx_runtime_1 = require("react/jsx-runtime");
var Nav_1 = __importDefault(require("Components/Nav/Nav"));
// import primary from 'Components/Navigation/Primary/primary'
var PrimaryNavigation_1 = __importDefault(require("Components/Navigation/Primary/PrimaryNavigation"));
var config_json_1 = require("Components/Navigation/Primary/config.json");
// import expand from '../../../scripts/expand'
var Primary_mdx_1 = __importDefault(require("./Primary/Primary.mdx"));
var Primary_context_1 = __importDefault(require("./Primary/Primary.context"));
// import PrimaryNavigationTwig from './Primary/Primary.example.twig'
var Template = function (args) {
    return (0, jsx_runtime_1.jsx)(PrimaryNavigation_1.default, __assign({}, args));
};
// const TwigTemplate: Story = () => {
//   React.useEffect(() => {
//     primary()
//     expand()
//   }, [])
//   return <div dangerouslySetInnerHTML={{ __html: PrimaryNavigationTwig({}) }}></div>
// }
exports.default = {
    component: PrimaryNavigation_1.default,
    subcomponents: { Nav: Nav_1.default },
    title: 'Design System/Components/Navigation/Primary Navigation',
    parameters: {
        docs: {
            description: {
                component: config_json_1.description,
            },
            page: Primary_mdx_1.default,
        },
        status: config_json_1.status,
    },
    argTypes: {
        attributes: {
            table: {
                disable: true,
            },
        },
    },
};
exports.primaryNavigation = Template.bind({});
exports.primaryNavigation.args = __assign({}, Primary_context_1.default.default.list);
