// @university-of-missouri/mizzou-design-system/dist/es5/stories/components/ContactCard/ContactCard.stories.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.contactCard = void 0;
var jsx_runtime_1 = require("react/jsx-runtime");
var ContactCard_1 = __importDefault(require("Components/ContactCard/ContactCard"));
var config_json_1 = require("Components/ContactCard/config.json");
var ContactCard_mdx_1 = __importDefault(require("./ContactCard.mdx"));
var Template = function (args) {
    return (0, jsx_runtime_1.jsx)(ContactCard_1.default, __assign({}, args));
};
exports.default = {
    component: ContactCard_1.default,
    title: 'Design System/Components/Contact Card',
    parameters: {
        docs: {
            description: {
                component: config_json_1.description,
            },
            page: ContactCard_mdx_1.default,
        },
        status: config_json_1.status,
    },
    argTypes: {
        title: {
            table: {
                category: 'Contact Information',
            },
        },
        subtitle: {
            table: {
                category: 'Contact Information',
            },
        },
        address: {
            table: {
                category: 'Contact Information',
            },
        },
        email: {
            table: {
                category: 'Contact Information',
            },
        },
        phoneNumber: {
            table: {
                category: 'Contact Information',
            },
        },
        faxNumber: {
            table: {
                category: 'Contact Information',
            },
        },
        attributes: {
            table: {
                disable: true,
            },
        },
        theme: {
            options: ['default', 'dark'],
            table: {
                category: 'Options',
            },
        },
        branded: {
            table: {
                category: 'Options',
            },
        },
        children: {
            table: {
                category: 'Options',
            },
        },
    },
};
exports.contactCard = Template.bind({});
exports.contactCard.args = {
    theme: 'default',
    branded: false,
    title: 'Contact Us',
    subtitle: 'Department Name',
    address: {
        street: '123 Building',
        locality: 'Columbia',
        region: 'MO',
        postalCode: '65211',
    },
};
