// @university-of-missouri/mizzou-design-system/src/stories/components/PersonProfile/PersonProfile.stories.tsx
import React from 'react'
import { Story } from '@storybook/react'

import { CardText } from 'Components/Card/Card'
import PersonProfile, { PersonProfileProps } from 'Components/PersonProfile/PersonProfile'
import { description, status } from 'Components/PersonProfile/config.json'
import mdx from './PersonProfile.mdx'

const Template: Story<PersonProfileProps> = (args) => {
  return <PersonProfile {...args} />
}

export default {
  component: PersonProfile,
  title: 'Design System/Components/Person Profile',
  parameters: {
    docs: {
      description: {
        component: description,
      },
      page: mdx,
    },
    status,
  },
  argTypes: {
    name: {
      table: {
        category: 'Props',
      },
    },
    position: {
      table: {
        category: 'Props',
      },
    },
    children: {
      table: {
        category: 'Props',
      },
    },
    attributes: {
      table: {
        disable: true,
      },
    },
    theme: {
      options: ['default', 'dark'],
      table: {
        category: 'Options',
      },
    },
    branded: {
      table: {
        category: 'Options',
      },
    },
    image: {
      table: {
        category: 'Options',
      },
    },
    links: {
      table: {
        category: 'Options',
      },
    },
    variant: {
      table: {
        category: 'Options',
      },
    },
  },
}
export const personProfile = Template.bind({})
personProfile.args = {
  name: 'Truman Tiger',
  position: 'Mascot',
  children: <CardText>Truman was acclaimed the Best Mascot in the Nation.</CardText>,
  image: {
    src: 'https://showme.missouri.edu/wp-content/uploads/2021/09/091321Truman1-600x600.jpg',
    alt: 'Truman Tiger',
  },
  variant: 'inline',
  theme: 'default',
  branded: false,
}
