// @university-of-missouri/mizzou-design-system/dist/es5/stories/components/Card/Card.stories.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CardGroup = exports.Default = void 0;
var jsx_runtime_1 = require("react/jsx-runtime");
var Card_1 = __importStar(require("Components/Card/Card"));
var config_json_1 = require("Components/Card/config.json");
var Card_mdx_1 = __importDefault(require("./Card.mdx"));
var Template = function (args) {
    return (0, jsx_runtime_1.jsx)(Card_1.default, __assign({}, args));
};
exports.default = {
    component: Card_1.default,
    subcomponents: {
        CardImage: Card_1.CardImage,
        CardTitle: Card_1.CardTitle,
        CardSubtitle: Card_1.CardSubtitle,
        CardText: Card_1.CardText,
        CardLinks: Card_1.CardLinks,
    },
    title: 'Design System/Components/Cards',
    parameters: {
        docs: {
            description: {
                component: config_json_1.description,
            },
            page: Card_mdx_1.default,
        },
        status: config_json_1.status,
    },
    argTypes: {
        children: {
            table: {
                category: 'Card Content',
            },
        },
        attributes: {
            table: {
                disable: true,
            },
        },
        theme: {
            options: ['default', 'dark'],
            table: {
                category: 'Options',
            },
        },
        variant: {
            options: ['default', 'outline', 'inline'],
            table: {
                category: 'Options',
            },
        },
        branded: {
            table: {
                category: 'Options',
            },
        },
    },
};
exports.Default = Template.bind({});
exports.Default.args = {
    theme: 'default',
    branded: false,
    children: ((0, jsx_runtime_1.jsxs)(jsx_runtime_1.Fragment, { children: [(0, jsx_runtime_1.jsx)(Card_1.CardImage, { src: "https://missouri.edu/sites/default/files/2020-07/CampusBeautyAerialOnJesse.jpg", alt: "Columns" }), (0, jsx_runtime_1.jsxs)(Card_1.CardBody, { children: [(0, jsx_runtime_1.jsx)(Card_1.CardTitle, { children: "Card Title" }), (0, jsx_runtime_1.jsx)(Card_1.CardSubtitle, { children: "Card Subtitle" }), (0, jsx_runtime_1.jsx)(Card_1.CardText, { children: "Card Description" }), (0, jsx_runtime_1.jsx)(Card_1.CardLinks, { links: [{ text: 'Learn more about this', href: 'http://missouri.edu/' }] })] })] })),
};
var CardGroup = function (args) { return ((0, jsx_runtime_1.jsxs)("div", __assign({ className: "miz-card-deck" }, { children: [(0, jsx_runtime_1.jsx)(Card_1.default, __assign({}, args)), (0, jsx_runtime_1.jsx)(Card_1.default, __assign({}, args)), (0, jsx_runtime_1.jsx)(Card_1.default, __assign({}, args))] }))); };
exports.CardGroup = CardGroup;
exports.CardGroup.args = {
    theme: 'default',
    branded: false,
    children: ((0, jsx_runtime_1.jsxs)(jsx_runtime_1.Fragment, { children: [(0, jsx_runtime_1.jsx)(Card_1.CardImage, { src: "https://missouri.edu/sites/default/files/2020-07/CampusBeautyAerialOnJesse.jpg", alt: "Columns" }), (0, jsx_runtime_1.jsxs)(Card_1.CardBody, { children: [(0, jsx_runtime_1.jsx)(Card_1.CardTitle, { children: "Card Title" }), (0, jsx_runtime_1.jsx)(Card_1.CardSubtitle, { children: "Card Subtitle" }), (0, jsx_runtime_1.jsx)(Card_1.CardText, { children: "Card Description" }), (0, jsx_runtime_1.jsx)(Card_1.CardLinks, { links: [{ text: 'Learn more about this', href: '#' }] })] })] })),
};
