// @university-of-missouri/mizzou-design-system/src/stories/components/Card/Card.stories.tsx
import React from 'react'
import { Story } from '@storybook/react'

import Card, {
  CardProps,
  CardImage,
  CardTitle,
  CardSubtitle,
  CardText,
  CardLinks,
  CardBody,
} from 'Components/Card/Card'
import { description, status } from 'Components/Card/config.json'
import mdx from './Card.mdx'

const Template: Story<CardProps> = (args) => {
  return <Card {...args} />
}

export default {
  component: Card,
  subcomponents: {
    CardImage,
    CardTitle,
    CardSubtitle,
    CardText,
    CardLinks,
  },
  title: 'Design System/Components/Cards',
  parameters: {
    docs: {
      description: {
        component: description,
      },
      page: mdx,
    },
    status,
  },
  argTypes: {
    children: {
      table: {
        category: 'Card Content',
      },
    },
    attributes: {
      table: {
        disable: true,
      },
    },
    theme: {
      options: ['default', 'dark'],
      table: {
        category: 'Options',
      },
    },
    variant: {
      options: ['default', 'outline', 'inline'],
      table: {
        category: 'Options',
      },
    },
    branded: {
      table: {
        category: 'Options',
      },
    },
  },
}
export const Default = Template.bind({})
Default.args = {
  theme: 'default',
  branded: false,
  children: (
    <>
      <CardImage
        src="https://missouri.edu/sites/default/files/2020-07/CampusBeautyAerialOnJesse.jpg"
        alt="Columns"
      />
      <CardBody>
        <CardTitle>Card Title</CardTitle>
        <CardSubtitle>Card Subtitle</CardSubtitle>
        <CardText>Card Description</CardText>
        <CardLinks links={[{ text: 'Learn more about this', href: 'http://missouri.edu/' }]} />
      </CardBody>
    </>
  ),
}

export const CardGroup: Story<CardProps> = (args) => (
  <div className="miz-card-deck">
    <Card {...args} />
    <Card {...args} />
    <Card {...args} />
  </div>
)
CardGroup.args = {
  theme: 'default',
  branded: false,
  children: (
    <>
      <CardImage
        src="https://missouri.edu/sites/default/files/2020-07/CampusBeautyAerialOnJesse.jpg"
        alt="Columns"
      />
      <CardBody>
        <CardTitle>Card Title</CardTitle>
        <CardSubtitle>Card Subtitle</CardSubtitle>
        <CardText>Card Description</CardText>
        <CardLinks links={[{ text: 'Learn more about this', href: '#' }]} />
      </CardBody>
    </>
  ),
}
