// @university-of-missouri/mizzou-design-system/dist/es5/stories/components/Ribbon/Ribbon.stories.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WithLinks = exports.Default = void 0;
var jsx_runtime_1 = require("react/jsx-runtime");
var Ribbon_1 = __importDefault(require("Components/Ribbon/Ribbon"));
var config_json_1 = require("Components/Ribbon/config.json");
var Ribbon_mdx_1 = __importDefault(require("./Ribbon.mdx"));
var Template = function (args) {
    return (0, jsx_runtime_1.jsx)(Ribbon_1.default, __assign({}, args));
};
exports.default = {
    component: Ribbon_1.default,
    title: 'Design System/Components/Ribbon',
    parameters: {
        docs: {
            description: {
                component: config_json_1.description,
            },
            page: Ribbon_mdx_1.default,
        },
        status: config_json_1.status,
    },
    argTypes: {
        attributes: {
            table: {
                disable: true,
            },
        },
        linkList: {
            table: {
                category: 'Options',
            },
        },
    },
};
exports.Default = Template.bind({});
exports.Default.args = {};
exports.WithLinks = Template.bind({});
exports.WithLinks.args = {
    linkList: [
        { text: 'List Item 1', url: '#' },
        { text: 'List Item 1', url: '#' },
        { text: 'List Item 1', url: '#' },
    ],
};
