// @university-of-missouri/mizzou-design-system/dist/es5/stories/components/Nav/Nav.stories.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Navs = void 0;
var jsx_runtime_1 = require("react/jsx-runtime");
var Nav_1 = __importDefault(require("Components/Nav/Nav"));
var config_json_1 = require("Components/Nav/config.json");
var Nav_context_1 = __importDefault(require("./Nav.context"));
var Nav_mdx_1 = __importDefault(require("./Nav.mdx"));
var Template = function (args) {
    return (0, jsx_runtime_1.jsx)(Nav_1.default, __assign({}, args));
};
exports.default = {
    component: Nav_1.default,
    title: 'Design System/Components/Navs',
    argTypes: {},
    parameters: {
        docs: {
            description: {
                component: config_json_1.description,
            },
            page: Nav_mdx_1.default,
        },
        status: config_json_1.status,
    },
};
exports.Navs = Template.bind({});
exports.Navs.args = __assign({}, Nav_context_1.default.default);
