// @university-of-missouri/mizzou-design-system/src/stories/patterns/Pagination/Pagination.stories.tsx
import React from 'react'
import { Story, Meta } from '@storybook/react'

import Pagination, { PaginationProps } from 'Components/Pagination/Pagination'
import { description, status } from 'Components/Pagination/config.json'
import mdx from './Pagination.mdx'

const StoryTemplate: Story<PaginationProps> = (args, { globals: { fileTypes } }) => {
  if (fileTypes === 'react') {
    return <Pagination {...args}></Pagination>
  }

  return <div>Null</div>
}

export default {
  component: Pagination,
  title: 'Design System/Patterns/Pagination',
  parameters: {
    docs: {
      description: {
        component: description,
      },
      page: mdx,
    },
    status,
  },
  argTypes: {
    children: {
      table: {
        category: 'Props',
      },
    },
    attributes: {
      table: {
        disable: true,
      },
    },
    branded: {
      table: {
        category: 'Options',
      },
    },
  },
} as Meta

export const pagination: Story = StoryTemplate.bind({})
pagination.args = {}
