// @university-of-missouri/mizzou-design-system/src/components/PageHeader/PageHeader.tsx
import * as React from 'react'
import DrupalAttribute from 'drupal-attribute'
import classNames from 'classnames'

export interface PageHeaderStyles {
  branded?: boolean
  variant?: string
  size?: string
}

export interface PageHeaderProps extends PageHeaderStyles {
  attributes: DrupalAttribute
  title: string
}

const PageHeader = ({
  attributes = new DrupalAttribute(''),
  variant = 'default',
  size = 'default',
  branded = false,
  title,
}: PageHeaderProps): JSX.Element => {
  const pageHeaderAttributes = {
    className: classNames(
      'miz-page-header',
      {
        [`miz-page-header--${size}`]: size !== 'default',
        'miz-page-header--dark': variant !== 'default',
        'miz-page-header--brand': branded,
      },
      attributes.get('class'),
    ),
  }

  attributes.forEach((value, key) => {
    if (key !== 'class') {
      pageHeaderAttributes[key] = value
    }
  })

  return (
    <header {...pageHeaderAttributes}>
      <div className="miz-container">
        <h2 className="miz-page-header__title">{title}</h2>
      </div>
    </header>
  )
}

export default PageHeader
