// @university-of-missouri/mizzou-design-system/src/stories/components/Layers/Events/Events.stories.tsx
import React from 'react'
import { Story } from '@storybook/react'

import Events, { EventsProps } from 'Components/Layers/Events/Events'
import { description, status } from 'Components/Layers/Events/config.json'
import mdx from './Events.mdx'

const Template: Story<EventsProps> = (args) => {
  return <Events {...args} />
}

export default {
  component: Events,
  title: 'Design System/Layers/Events',
  parameters: {
    docs: {
      description: {
        component: description,
      },
      page: mdx,
    },
    status,
  },
  argTypes: {
    attributes: {
      table: {
        disable: true,
      },
    },
    title: {
      table: {
        category: 'Layer Options',
      },
    },
    kicker: {
      table: {
        category: 'Layer Options',
      },
    },
    buttons: {
      table: {
        category: 'Layer Options',
      },
    },
    method: {
      table: {
        category: 'Calendar Options',
      },
    },
    term: {
      table: {
        category: 'Calendar Options',
      },
    },
    count: {
      control: {
        type: 'range',
        min: 1,
        max: 4,
        step: 1,
      },
      table: {
        category: 'Calendar Options',
      },
    },
  },
}
export const events = Template.bind({})
events.args = {
  title: 'Events Section Title',
  kicker: 'Mizzou',
  method: 'search',
  term: 'education',
  buttons: [{ title: 'More Events', link: 'https://calendar.missouri.edu/' }],
}
