// @university-of-missouri/mizzou-design-system/dist/es5/stories/components/Layers/SplitLayout/SplitLayout.stories.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.splitLayout = void 0;
var jsx_runtime_1 = require("react/jsx-runtime");
var SplitLayout_1 = __importDefault(require("Components/Layers/SplitLayout/SplitLayout"));
var config_json_1 = require("Components/Layers/SplitLayout/config.json");
var SplitLayout_mdx_1 = __importDefault(require("./SplitLayout.mdx"));
var Template = function (args) { return (0, jsx_runtime_1.jsx)(SplitLayout_1.default, __assign({}, args)); };
exports.default = {
    component: SplitLayout_1.default,
    title: 'Design System/Layers/Split Layout',
    parameters: {
        docs: {
            description: {
                component: config_json_1.description,
            },
            page: SplitLayout_mdx_1.default,
        },
        status: config_json_1.status,
    },
    argTypes: {
        title: {
            table: {
                category: 'Props',
            },
        },
        kicker: {
            table: {
                category: 'Props',
            },
        },
        children: {
            table: {
                category: 'Props',
            },
        },
        attributes: {
            table: {
                disable: true,
            },
        },
        variant: {
            options: ['default', 'black', 'image', 'offset'],
            table: {
                category: 'Options',
            },
        },
        align: {
            table: {
                category: 'Options',
            },
        },
        image: {
            table: {
                category: 'Options',
            },
        },
        backgroundImage: {
            table: {
                category: 'Options',
            },
        },
        buttons: {
            table: {
                category: 'Options',
            },
        },
    },
};
exports.splitLayout = Template.bind({});
exports.splitLayout.args = {
    title: 'Split Layout Title',
    kicker: 'Split Layout Kicker',
    children: (0, jsx_runtime_1.jsxs)("p", { children: ["Looking for a vibrant community offering endless opportunities to learn and grow? ", (0, jsx_runtime_1.jsx)("a", __assign({ href: "//missouri.edu/" }, { children: "It\u2019s always a great time to be a Tiger" })), ". Schedule a campus tour or try a virtual chat with current students to discover why you\u2019ll love MU."] }),
    variant: 'black',
    align: 'left',
    image: 'https://missouri.edu/sites/default/files/2020-07/CampusBeautyAerialOnJesse.jpg',
    buttons: [{ title: 'Split Layout Button', link: 'https://missouri.edu/' }],
};
