// @university-of-missouri/mizzou-design-system/src/stories/components/NewsCard/NewsCard.stories.tsx
import React from 'react'
import { Story } from '@storybook/react'

// import { CardProps } from 'Components/Card/Card'
import NewsCard, { NewsCardProps } from 'Components/NewsCard/NewsCard'
import { description, status } from 'Components/NewsCard/config.json'
import mdx from './NewsCard.mdx'

const Template: Story<NewsCardProps> = (args) => {
  return <NewsCard {...args} />
}

export default {
  component: NewsCard,
  title: 'Design System/Components/News Cards',
  parameters: {
    docs: {
      description: {
        component: description,
      },
      page: mdx,
    },
    status,
  },
  argTypes: {
    title: {
      table: {
        category: 'Card Content',
      },
    },
    subtitle: {
      table: {
        category: 'Card Content',
      },
    },
    topic: {
      table: {
        category: 'Card Content',
      },
    },
    image: {
      table: {
        category: 'Card Content',
      },
    },
    links: {
      table: {
        category: 'Card Content',
      },
    },
    children: {
      table: {
        category: 'Card Content',
      },
    },
    attributes: {
      table: {
        disable: true,
      },
    },
    variant: {
      table: {
        category: 'Options',
      },
    },
    theme: {
      options: ['default', 'dark'],
      table: {
        category: 'Options',
      },
    },
    branded: {
      table: {
        category: 'Options',
      },
    },
  },
}
export const Default = Template.bind({})
Default.args = {
  title: 'Mizzou on the moon',
  subtitle: 'Card Subtitle',
  topic: 'Mizzou Made',
  children:
    // eslint-disable-next-line prettier/prettier
    '50th anniversary of the moon landing shines spotlight on MU\'s ties to the space program.',
  image: 'https://missouri.edu/sites/default/files/2020-07/CampusBeautyAerialOnJesse.jpg',
  links: [{ text: 'Learn more about this', href: '#' }],
  theme: 'default',
  branded: false,
}

export const CardGroup: Story<NewsCardProps> = (args) => (
  <div className="miz-card-deck">
    <NewsCard {...args} />
    <NewsCard {...args} />
    <NewsCard {...args} />
  </div>
)
CardGroup.args = {
  title: 'Mizzou on the moon',
  subtitle: 'Card Subtitle',
  topic: 'Mizzou Made',
  children:
    // eslint-disable-next-line prettier/prettier
    '50th anniversary of the moon landing shines spotlight on MU\'s ties to the space program.',
  image: 'https://missouri.edu/sites/default/files/2020-07/CampusBeautyAerialOnJesse.jpg',
  links: [{ text: 'Learn more about this', href: '#' }],
  theme: 'default',
  branded: false,
}
