// @university-of-missouri/mizzou-design-system/src/stories/components/LinkList/LinkList.stories.tsx
import React from 'react'
import { Story } from '@storybook/react'

import LinkList, { LinkListProps } from 'Components/LinkList/LinkList'
import { description, status } from 'Components/LinkList/config.json'
import mdx from './LinkList.mdx'

const Template: Story<LinkListProps> = (args) => {
  return <LinkList {...args} />
}

export default {
  component: LinkList,
  title: 'Design System/Components/Link Lists',
  parameters: {
    docs: {
      description: {
        component: description,
      },
      page: mdx,
    },
    status,
  },
  argTypes: {
    attributes: {
      table: {
        disable: true,
      },
    },
    layout: {
      table: {
        category: 'Options',
      },
    },
    align: {
      table: {
        category: 'Options',
      },
    },
    fill: {
      table: {
        category: 'Options',
      },
    },
    branded: {
      table: {
        category: 'Options',
      },
    },
    list: {
      table: {
        category: 'Options',
      },
    },
  },
}
export const linkLists = Template.bind({})
linkLists.args = {
  layout: 'default',
  align: 'default',
  fill: false,
  branded: false,
  list: [
    { text: 'List Item 1', url: '#' },
    { text: 'List Item 2', url: '#' },
    { text: 'List Item 3', url: '#' },
  ],
}

// export const Inline = Template.bind({})
// Inline.args = {
//   layout: 'inline',
//   align: 'default',
//   fill: false,
//   branded: false,
//   list: [
//     { text: 'List Item 1', url: '#' },
//     { text: 'List Item 1', url: '#' },
//     { text: 'List Item 1', url: '#' },
//   ],
// }
