// @university-of-missouri/mizzou-design-system/src/stories/components/ContactCard/ContactCard.stories.tsx
import React from 'react'
import { Story } from '@storybook/react'

import ContactCard, { ContactCardProps } from 'Components/ContactCard/ContactCard'
import { description, status } from 'Components/ContactCard/config.json'
import mdx from './ContactCard.mdx'

const Template: Story<ContactCardProps> = (args) => {
  return <ContactCard {...args} />
}

export default {
  component: ContactCard,
  title: 'Design System/Components/Contact Card',
  parameters: {
    docs: {
      description: {
        component: description,
      },
      page: mdx,
    },
    status,
  },
  argTypes: {
    title: {
      table: {
        category: 'Contact Information',
      },
    },
    subtitle: {
      table: {
        category: 'Contact Information',
      },
    },
    address: {
      table: {
        category: 'Contact Information',
      },
    },
    email: {
      table: {
        category: 'Contact Information',
      },
    },
    phoneNumber: {
      table: {
        category: 'Contact Information',
      },
    },
    faxNumber: {
      table: {
        category: 'Contact Information',
      },
    },
    attributes: {
      table: {
        disable: true,
      },
    },
    theme: {
      options: ['default', 'dark'],
      table: {
        category: 'Options',
      },
    },
    branded: {
      table: {
        category: 'Options',
      },
    },
    children: {
      table: {
        category: 'Options',
      },
    },
  },
}
export const contactCard = Template.bind({})
contactCard.args = {
  theme: 'default',
  branded: false,
  title: 'Contact Us',
  subtitle: 'Department Name',
  address: {
    street: '123 Building',
    locality: 'Columbia',
    region: 'MO',
    postalCode: '65211',
  },
}
