// @university-of-missouri/mizzou-design-system/src/components/Footer/Footer.tsx
import * as React from 'react'
import DrupalAttribute from 'drupal-attribute'
import classNames from 'classnames'

import ContactCard from 'Components/ContactCard/ContactCard'
import Colophon from 'Components/Colophon/Colophon'
import Signature, { Logo, Wordmark } from 'Components/Signature/Signature'

export interface FooterProps {
  attributes: DrupalAttribute
  theme: 'light' | 'dark'
  brandContainer: React.ReactNode
  infoContainer: React.ReactNode
  eeoaaContainer: React.ReactNode
  colophonContainer: React.ReactNode
}

const Footer = ({
  attributes = new DrupalAttribute(''),
  theme = 'light',
  brandContainer = (
    <Signature size="medium" reverse={theme === 'dark'}>
      <Logo />
      <Wordmark />
    </Signature>
  ),
  infoContainer = <ContactCard title="Contact Us" theme={theme === 'light' ? 'default' : 'dark'} />,
  eeoaaContainer = (
    <small>
      MU is an{' '}
      <a href="//missouri.edu/eeo-aa/">
        {' '}
        equal opportunity/access/affirmative action/pro-disabled and veteran employer
      </a>{' '}
      and does not discriminate on the basis of sex in our education programs or activities,
      pursuant to Title IX and 34 CFR Part 106. For more information, visit{' '}
      <a href="//missouri.edu/eeo-aa/">MU’s Nondiscrimination Policy</a> or the{' '}
      <a href="https://equity.missouri.edu/education-prevention-and-outreach/student-training-not-anymore/">
        Office of Institutional Equity
      </a>
      .
    </small>
  ),
  colophonContainer = <Colophon />,
}: FooterProps): JSX.Element => {
  const FooterAttributes = {
    className: classNames(
      'miz-footer',
      {
        'miz-footer--dark': theme === 'dark',
      },
      attributes.get('class'),
    ),
  }

  attributes.forEach((value, key) => {
    if (key !== 'class') {
      FooterAttributes[key] = value
    }
  })

  return (
    <footer {...FooterAttributes}>
      <div className="miz-footer__layer">
        <div className="miz-footer__brand">
          <div className="miz-container">{brandContainer}</div>
        </div>
      </div>

      <div className="miz-footer__layer">
        <div className="miz-footer__information">{infoContainer}</div>
      </div>

      <div className="miz-footer__layer">
        <div className="miz-container">
          <div className="miz-footer__eeoaa">
            <p>{eeoaaContainer}</p>
          </div>
        </div>
      </div>

      <div className="miz-footer__layer miz-fill--black">
        <div className="miz-container">
          <div className="miz-footer__colophon">{colophonContainer}</div>
        </div>
      </div>
    </footer>
  )
}

export default Footer
