// @wordpress/block-library/build/footnotes/edit.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = FootnotesEdit;
var _element = require("@wordpress/element");
var _blockEditor = require("@wordpress/block-editor");
var _coreData = require("@wordpress/core-data");
var _i18n = require("@wordpress/i18n");
var _components = require("@wordpress/components");
var _icons = require("@wordpress/icons");
/**
 * WordPress dependencies
 */

function FootnotesEdit({
  context: {
    postType,
    postId
  }
}) {
  const [meta, updateMeta] = (0, _coreData.useEntityProp)('postType', postType, 'meta', postId);
  const footnotes = meta?.footnotes ? JSON.parse(meta.footnotes) : [];
  const blockProps = (0, _blockEditor.useBlockProps)();
  if (postType !== 'post' && postType !== 'page') {
    return (0, _element.createElement)("div", {
      ...blockProps
    }, (0, _element.createElement)(_components.Placeholder, {
      icon: (0, _element.createElement)(_blockEditor.BlockIcon, {
        icon: _icons.formatListNumbered
      }),
      label: (0, _i18n.__)('Footnotes')
      // To do: add instructions. We can't add new string in RC.
    }));
  }

  if (!footnotes.length) {
    return (0, _element.createElement)("div", {
      ...blockProps
    }, (0, _element.createElement)(_components.Placeholder, {
      icon: (0, _element.createElement)(_blockEditor.BlockIcon, {
        icon: _icons.formatListNumbered
      }),
      label: (0, _i18n.__)('Footnotes'),
      instructions: (0, _i18n.__)('Footnotes found in blocks within this document will be displayed here.')
    }));
  }
  return (0, _element.createElement)("ol", {
    ...blockProps
  }, footnotes.map(({
    id,
    content
  }) => (0, _element.createElement)("li", {
    key: id
  }, (0, _element.createElement)(_blockEditor.RichText, {
    id: id,
    tagName: "span",
    value: content,
    identifier: id
    // To do: figure out why the browser is not scrolling
    // into view when it receives focus.
    ,
    onFocus: event => {
      if (!event.target.textContent.trim()) {
        event.target.scrollIntoView();
      }
    },
    onChange: nextFootnote => {
      updateMeta({
        ...meta,
        footnotes: JSON.stringify(footnotes.map(footnote => {
          return footnote.id === id ? {
            content: nextFootnote,
            id
          } : footnote;
        }))
      });
    }
  }), ' ', (0, _element.createElement)("a", {
    href: `#${id}-link`
  }, "\u21A9\uFE0E"))));
}
