// @university-of-missouri/mizzou-design-system/dist/es5/stories/components/ListGroup/ListGroup.stories.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.listGroup = void 0;
var jsx_runtime_1 = require("react/jsx-runtime");
var ListGroup_1 = __importDefault(require("Components/ListGroup/ListGroup"));
var config_json_1 = require("Components/ListGroup/config.json");
var ListGroup_mdx_1 = __importDefault(require("./ListGroup.mdx"));
var Template = function (args) {
    return (0, jsx_runtime_1.jsx)(ListGroup_1.default, __assign({}, args));
};
exports.default = {
    component: ListGroup_1.default,
    title: 'Design System/Components/List Group',
    parameters: {
        docs: {
            description: {
                component: config_json_1.description,
            },
            page: ListGroup_mdx_1.default,
        },
        status: config_json_1.status,
    },
    argTypes: {
        attributes: {
            table: {
                disable: true,
            },
        },
    },
};
exports.listGroup = Template.bind({});
exports.listGroup.args = {
    type: 'unordered',
    list: {
        items: [
            {
                item: {
                    content: 'Item 1',
                },
            },
            {
                item: {
                    content: 'Item 2',
                },
            },
            {
                item: {
                    content: 'Item 3',
                },
            },
        ],
    },
};
