// @university-of-missouri/eslint-config/.eslintrc.js
/**
 * @format
 */
module.exports = {
  env: {
    'shared-node-browser': true,
    es2021: true,
  },
  globals: {
    Atomics: 'readonly',
    SharedArrayBuffer: 'readonly',
    module: true,
    require: true,
    __dirname: true,
    process: true,
    console: true,
  },
  parserOptions: {
    ecmaVersion: 2021,
    sourceType: 'module',
  },
  extends: [
    'airbnb-base', // https://github.com/airbnb/javascript/tree/master/packages/eslint-config-airbnb-base/rules
  ],
  overrides: [
    // Plain JS files
    {
      files: ['**/*.js'],
      extends: [require.resolve('./rules/base.js'), 'prettier'],
      plugins: ['prettier'],
      rules: {
        'prettier/prettier': 'error',
      },
    },
    // React files (.jsx)
    {
      files: ['**/*.jsx'],
      extends: [
        'plugin:react/recommended',
        require.resolve('./rules/base.js'),
        require.resolve('./rules/react.js'),
        'prettier',
      ],
      plugins: ['prettier', 'react'],
      parserOptions: {
        ecmaFeatures: {
          jsx: true,
        },
        sourceType: 'module',
      },
      rules: {
        'prettier/prettier': 'error',
      },
      settings: {
        react: { version: '18.2.0' },
      },
    },
    // TypeScript
    {
      files: ['**/*.ts', '**/*.tsx'],
      parser: '@typescript-eslint/parser',
      plugins: ['@typescript-eslint', 'react', 'import'],
      extends: [
        'eslint:recommended',
        'plugin:react/recommended',
        'plugin:@typescript-eslint/eslint-recommended',
        'plugin:@typescript-eslint/recommended',
      ],
      settings: {
        'import/parsers': {
          '@typescript-eslint/parser': ['.ts', '.tsx'],
        },
        'import/resolver': {
          typescript: {
            alwaysTryTypes: true,
          },
        },
        react: { version: '17.0.2' },
      },
      env: { browser: true, es6: true, node: true },
      globals: { Atomics: 'readonly', SharedArrayBuffer: 'readonly' },
      parserOptions: {
        ecmaFeatures: { jsx: true },
        ecmaVersion: 2021,
        sourceType: 'module',
        project: true,
      },
      rules: {
        indent: ['error', 2, { SwitchCase: 1 }],
        'linebreak-style': ['error', 'unix'],
        quotes: 0,
        semi: 0,
        '@typescript-eslint/quotes': [
          'error',
          'single',
          { avoidEscape: false, allowTemplateLiterals: true },
        ],
        'comma-dangle': ['error', 'always-multiline'],
        '@typescript-eslint/no-explicit-any': 0,
        'no-use-before-define': 'off',
        '@typescript-eslint/no-use-before-define': ['error'],
        'import/no-unresolved': 0,
        'import/extensions': [0],
        'import/no-extraneous-dependencies': [
          'error',
          {
            devDependencies: true,
            optionalDependencies: true,
            peerDependencies: true,
            bundledDependencies: true,
          },
        ],
      },
    },
  ],
}
