// @university-of-missouri/mizzou-design-system/dist/es5/stories/components/Header/Header.stories.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.header = void 0;
var jsx_runtime_1 = require("react/jsx-runtime");
var Header_1 = __importDefault(require("Components/Header/Header"));
var config_json_1 = require("Components/Header/config.json");
var Header_mdx_1 = __importDefault(require("./Header.mdx"));
var Template = function (args) {
    return (0, jsx_runtime_1.jsx)(Header_1.default, __assign({}, args));
};
exports.default = {
    component: Header_1.default,
    title: 'Design System/Components/Header',
    parameters: {
        docs: {
            description: {
                component: config_json_1.description,
            },
            page: Header_mdx_1.default,
        },
        status: config_json_1.status,
    },
    argTypes: {
        attributes: {
            table: {
                disable: true,
            },
        },
        theme: {
            options: ['light', 'dark'],
            table: {
                category: 'Options',
            },
        },
    },
};
exports.header = Template.bind({});
exports.header.args = {
    theme: 'light',
};
