// @university-of-missouri/mizzou-design-system/dist/es5/stories/components/Button/Button.stories.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.KitchenSink = exports.Buttons = void 0;
var jsx_runtime_1 = require("react/jsx-runtime");
var drupal_attribute_1 = __importDefault(require("drupal-attribute"));
var Button_1 = __importDefault(require("Components/Button/Button"));
var config_json_1 = require("Components/Button/config.json");
var Button_mdx_1 = __importDefault(require("./Button.mdx"));
var KitchenSink_mdx_1 = __importDefault(require("./KitchenSink.mdx"));
var ButtonStoryTemplate = function (_a) {
    var _b = _a.state, state = _b === void 0 ? false : _b, _c = _a.styleContext, styleContext = _c === void 0 ? '' : _c, args = __rest(_a, ["state", "styleContext"]);
    var props = args;
    var stateStyling = new drupal_attribute_1.default('');
    if (state) {
        stateStyling.addClass(state).setAttribute('tabIndex', '1');
    }
    if (typeof styleContext === 'string') {
        stateStyling.addClass(styleContext);
    }
    return (0, jsx_runtime_1.jsx)(Button_1.default, __assign({}, props, { attributes: stateStyling }));
};
exports.default = {
    component: Button_1.default,
    title: 'Design System/Components/Buttons',
    parameters: {
        docs: {
            description: {
                component: config_json_1.description,
            },
            page: Button_mdx_1.default,
        },
        status: config_json_1.status,
    },
    argTypes: {
        children: {
            table: {
                category: 'Props',
            },
        },
        attributes: {
            table: {
                disable: true,
            },
        },
        onClick: {
            table: {
                disable: true,
            },
        },
        primary: {
            table: {
                category: 'Options',
            },
        },
        secondary: {
            table: {
                category: 'Options',
            },
        },
        branded: {
            table: {
                category: 'Options',
            },
        },
        variant: {
            table: {
                category: 'Options',
            },
        },
        size: {
            table: {
                category: 'Options',
            },
        },
    },
};
exports.Buttons = ButtonStoryTemplate.bind({});
exports.Buttons.args = {
    children: 'Mizzou Button',
    primary: false,
    secondary: false,
    branded: false,
    size: 'default',
    variant: 'default',
};
exports.KitchenSink = ButtonStoryTemplate.bind({});
exports.KitchenSink.storyName = 'Sink';
exports.KitchenSink.parameters = {
    docs: {
        page: KitchenSink_mdx_1.default,
        previewTabs: {
            canvas: { hidden: true },
        },
    },
};
