// @university-of-missouri/mizzou-design-system/dist/es5/stories/components/Layers/Events/Events.stories.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.events = void 0;
var jsx_runtime_1 = require("react/jsx-runtime");
var Events_1 = __importDefault(require("Components/Layers/Events/Events"));
var config_json_1 = require("Components/Layers/Events/config.json");
var Events_mdx_1 = __importDefault(require("./Events.mdx"));
var Template = function (args) {
    return (0, jsx_runtime_1.jsx)(Events_1.default, __assign({}, args));
};
exports.default = {
    component: Events_1.default,
    title: 'Design System/Layers/Events',
    parameters: {
        docs: {
            description: {
                component: config_json_1.description,
            },
            page: Events_mdx_1.default,
        },
        status: config_json_1.status,
    },
    argTypes: {
        attributes: {
            table: {
                disable: true,
            },
        },
        title: {
            table: {
                category: 'Layer Options',
            },
        },
        kicker: {
            table: {
                category: 'Layer Options',
            },
        },
        buttons: {
            table: {
                category: 'Layer Options',
            },
        },
        method: {
            table: {
                category: 'Calendar Options',
            },
        },
        term: {
            table: {
                category: 'Calendar Options',
            },
        },
        count: {
            control: {
                type: 'range',
                min: 1,
                max: 4,
                step: 1,
            },
            table: {
                category: 'Calendar Options',
            },
        },
    },
};
exports.events = Template.bind({});
exports.events.args = {
    title: 'Events Section Title',
    kicker: 'Mizzou',
    method: 'search',
    term: 'education',
    buttons: [{ title: 'More Events', link: 'https://calendar.missouri.edu/' }],
};
