// @university-of-missouri/mizzou-design-system/src/stories/components/Layers/LinkList/LinkListLayer.stories.tsx
import React from 'react'
import { Story } from '@storybook/react'

import LinkListLayer, { LinkListLayerProps } from 'Components/Layers/LinkList/LinkList'
import { description, status } from 'Components/Layers/LinkList/config.json'
import mdx from './LinkList.mdx'

const Template: Story<LinkListLayerProps> = (args) => {
  return <LinkListLayer {...args} />
}

export default {
  component: LinkListLayer,
  title: 'Design System/Layers/Link List',
  parameters: {
    docs: {
      description: {
        component: description,
      },
      page: mdx,
    },
    status,
  },
  argTypes: {
    attributes: {
      table: {
        disable: true,
      },
    },
    layout: {
      table: {
        category: 'Options',
        disable: true,
      },
    },
    align: {
      table: {
        category: 'Options',
        disable: true,
      },
    },
    fill: {
      table: {
        category: 'Options',
        disable: true,
      },
    },
    branded: {
      table: {
        category: 'Options',
        disable: true,
      },
    },
    list: {
      table: {
        category: 'Options',
      },
    },
  },
}

export const linkList = Template.bind({})
linkList.args = {
  title: 'Link List Title',
  list: [
    { text: 'List Item 1', url: '#' },
    { text: 'List Item 2', url: '#' },
    { text: 'List Item 3', url: '#' },
  ],
}
