// @university-of-missouri/mizzou-design-system/dist/es5/stories/components/NewsCard/NewsCard.stories.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CardGroup = exports.Default = void 0;
var jsx_runtime_1 = require("react/jsx-runtime");
// import { CardProps } from 'Components/Card/Card'
var NewsCard_1 = __importDefault(require("Components/NewsCard/NewsCard"));
var config_json_1 = require("Components/NewsCard/config.json");
var NewsCard_mdx_1 = __importDefault(require("./NewsCard.mdx"));
var Template = function (args) {
    return (0, jsx_runtime_1.jsx)(NewsCard_1.default, __assign({}, args));
};
exports.default = {
    component: NewsCard_1.default,
    title: 'Design System/Components/News Cards',
    parameters: {
        docs: {
            description: {
                component: config_json_1.description,
            },
            page: NewsCard_mdx_1.default,
        },
        status: config_json_1.status,
    },
    argTypes: {
        title: {
            table: {
                category: 'Card Content',
            },
        },
        subtitle: {
            table: {
                category: 'Card Content',
            },
        },
        topic: {
            table: {
                category: 'Card Content',
            },
        },
        image: {
            table: {
                category: 'Card Content',
            },
        },
        links: {
            table: {
                category: 'Card Content',
            },
        },
        children: {
            table: {
                category: 'Card Content',
            },
        },
        attributes: {
            table: {
                disable: true,
            },
        },
        variant: {
            table: {
                category: 'Options',
            },
        },
        theme: {
            options: ['default', 'dark'],
            table: {
                category: 'Options',
            },
        },
        branded: {
            table: {
                category: 'Options',
            },
        },
    },
};
exports.Default = Template.bind({});
exports.Default.args = {
    title: 'Mizzou on the moon',
    subtitle: 'Card Subtitle',
    topic: 'Mizzou Made',
    children: 
    // eslint-disable-next-line prettier/prettier
    '50th anniversary of the moon landing shines spotlight on MU\'s ties to the space program.',
    image: 'https://missouri.edu/sites/default/files/2020-07/CampusBeautyAerialOnJesse.jpg',
    links: [{ text: 'Learn more about this', href: '#' }],
    theme: 'default',
    branded: false,
};
var CardGroup = function (args) { return ((0, jsx_runtime_1.jsxs)("div", __assign({ className: "miz-card-deck" }, { children: [(0, jsx_runtime_1.jsx)(NewsCard_1.default, __assign({}, args)), (0, jsx_runtime_1.jsx)(NewsCard_1.default, __assign({}, args)), (0, jsx_runtime_1.jsx)(NewsCard_1.default, __assign({}, args))] }))); };
exports.CardGroup = CardGroup;
exports.CardGroup.args = {
    title: 'Mizzou on the moon',
    subtitle: 'Card Subtitle',
    topic: 'Mizzou Made',
    children: 
    // eslint-disable-next-line prettier/prettier
    '50th anniversary of the moon landing shines spotlight on MU\'s ties to the space program.',
    image: 'https://missouri.edu/sites/default/files/2020-07/CampusBeautyAerialOnJesse.jpg',
    links: [{ text: 'Learn more about this', href: '#' }],
    theme: 'default',
    branded: false,
};
