// @university-of-missouri/mizzou-design-system/dist/es5/stories/components/Accordion/Accordion.stories.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.accordion = void 0;
var jsx_runtime_1 = require("react/jsx-runtime");
var react_1 = __importDefault(require("react"));
var drupal_attribute_1 = __importDefault(require("drupal-attribute"));
var Accordion_1 = __importDefault(require("Components/Accordion/Accordion"));
var config_json_1 = require("Components/Accordion/config.json");
var Accordion_example_twig_1 = __importDefault(require("./Accordion.example.twig"));
var expand_1 = __importDefault(require("../../../scripts/expand"));
var Accordion_mdx_1 = __importDefault(require("./Accordion.mdx"));
var StoryTemplate = function (args, _a) {
    var fileTypes = _a.globals.fileTypes;
    if (fileTypes === 'react') {
        return ((0, jsx_runtime_1.jsxs)(jsx_runtime_1.Fragment, { children: [(0, jsx_runtime_1.jsx)(Accordion_1.default, __assign({}, args)), (0, jsx_runtime_1.jsx)(Accordion_1.default, __assign({}, args))] }));
    }
    react_1.default.useEffect(function () {
        (0, expand_1.default)();
    }, []);
    return (0, jsx_runtime_1.jsx)("div", { dangerouslySetInnerHTML: { __html: (0, Accordion_example_twig_1.default)({}) } });
};
exports.default = {
    component: Accordion_1.default,
    title: 'Design System/Components/Accordion',
    parameters: {
        docs: {
            description: {
                component: config_json_1.description,
            },
            page: Accordion_mdx_1.default,
        },
        status: config_json_1.status,
    },
    argTypes: {
        children: {
            table: {
                category: 'Props',
            },
        },
        attributes: {
            table: {
                disable: true,
            },
        },
        branded: {
            table: {
                category: 'Options',
            },
        },
    },
};
exports.accordion = StoryTemplate.bind({});
exports.accordion.args = {
    attributes: new drupal_attribute_1.default('').setAttribute('id', 'testAccordion'),
    branded: true,
    theme: 'default',
    button: 'Accordion Button',
    content: ((0, jsx_runtime_1.jsxs)(jsx_runtime_1.Fragment, { children: [(0, jsx_runtime_1.jsx)("h3", { children: "Section Header" }), (0, jsx_runtime_1.jsx)("p", { children: "Section Content" })] })),
};
