// @wordpress/block-library/build-module/page-list/convert-to-links-modal.js
import { createElement } from "@wordpress/element";
/**
 * WordPress dependencies
 */
import { Button, Modal } from '@wordpress/components';
import { __ } from '@wordpress/i18n';
export const convertDescription = __('This page list is synced with the published pages on your site. Detach the page list to add, delete, or reorder pages yourself.');
export function ConvertToLinksModal({
  onClick,
  onClose,
  disabled
}) {
  return createElement(Modal, {
    onRequestClose: onClose,
    title: __('Edit Page List'),
    className: 'wp-block-page-list-modal',
    aria: {
      describedby: 'wp-block-page-list-modal__description'
    }
  }, createElement("p", {
    id: 'wp-block-page-list-modal__description'
  }, convertDescription), createElement("div", {
    className: "wp-block-page-list-modal-buttons"
  }, createElement(Button, {
    variant: "tertiary",
    onClick: onClose
  }, __('Cancel')), createElement(Button, {
    variant: "primary",
    disabled: disabled,
    onClick: onClick
  }, __('Detach'))));
}
