// @university-of-missouri/mizzou-design-system/src/stories/components/Accordion/Accordion.stories.tsx
import React from 'react'
import { Story, Meta } from '@storybook/react'
import DrupalAttribute from 'drupal-attribute'

import Accordion, { AccordionProps } from 'Components/Accordion/Accordion'
import { description, status } from 'Components/Accordion/config.json'
import AccordionTwig from './Accordion.example.twig'
import expand from '../../../scripts/expand'
import mdx from './Accordion.mdx'

const StoryTemplate: Story<AccordionProps> = (args, { globals: { fileTypes } }) => {
  if (fileTypes === 'react') {
    return (
      <>
        <Accordion {...args}></Accordion>
        <Accordion {...args}></Accordion>
      </>
    )
  }

  React.useEffect(() => {
    expand()
  }, [])
  return <div dangerouslySetInnerHTML={{ __html: AccordionTwig({}) }}></div>
}

export default {
  component: Accordion,
  title: 'Design System/Components/Accordion',
  parameters: {
    docs: {
      description: {
        component: description,
      },
      page: mdx,
    },
    status,
  },
  argTypes: {
    children: {
      table: {
        category: 'Props',
      },
    },
    attributes: {
      table: {
        disable: true,
      },
    },
    branded: {
      table: {
        category: 'Options',
      },
    },
  },
} as Meta

export const accordion: Story = StoryTemplate.bind({})
accordion.args = {
  attributes: new DrupalAttribute('').setAttribute('id', 'testAccordion'),
  branded: true,
  theme: 'default',
  button: 'Accordion Button',
  content: (
    <>
      <h3>Section Header</h3>
      <p>Section Content</p>
    </>
  ),
}
