// @wordpress/block-library/build/code/transforms.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _blocks = require("@wordpress/blocks");
/**
 * WordPress dependencies
 */

const transforms = {
  from: [{
    type: 'enter',
    regExp: /^```$/,
    transform: () => (0, _blocks.createBlock)('core/code')
  }, {
    type: 'block',
    blocks: ['core/html', 'core/paragraph'],
    transform: ({
      content
    }) => {
      return (0, _blocks.createBlock)('core/code', {
        content
      });
    }
  }, {
    type: 'raw',
    isMatch: node => node.nodeName === 'PRE' && node.children.length === 1 && node.firstChild.nodeName === 'CODE',
    schema: {
      pre: {
        children: {
          code: {
            children: {
              '#text': {}
            }
          }
        }
      }
    }
  }],
  to: [{
    type: 'block',
    blocks: ['core/paragraph'],
    transform: ({
      content
    }) => {
      return (0, _blocks.createBlock)('core/paragraph', {
        content: content.replace(/\n/g, '<br>')
      });
    }
  }]
};
var _default = transforms;
exports.default = _default;
