// @university-of-missouri/mizzou-design-system/dist/es5/stories/components/Layers/Quote/Quote.stories.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.quote = void 0;
var jsx_runtime_1 = require("react/jsx-runtime");
var Quote_1 = __importDefault(require("Components/Layers/Quote/Quote"));
var config_json_1 = require("Components/Layers/Quote/config.json");
var Quote_mdx_1 = __importDefault(require("./Quote.mdx"));
var Template = function (args) {
    return (0, jsx_runtime_1.jsx)(Quote_1.default, __assign({}, args));
};
exports.default = {
    component: Quote_1.default,
    title: 'Design System/Layers/Quote',
    parameters: {
        docs: {
            description: {
                component: config_json_1.description,
            },
            page: Quote_mdx_1.default,
        },
        status: config_json_1.status,
    },
    argTypes: {
        title: {
            table: {
                category: 'Props',
            },
        },
        kicker: {
            table: {
                category: 'Props',
            },
        },
        attributes: {
            table: {
                disable: true,
            },
        },
        attribution: {
            table: {
                category: 'Options',
            },
        },
        degree: {
            table: {
                category: 'Options',
            },
        },
        children: {
            table: {
                category: 'Options',
            },
        },
        image: {
            table: {
                category: 'Options',
            },
        },
        buttons: {
            table: {
                category: 'Options',
            },
        },
    },
};
exports.quote = Template.bind({});
exports.quote.args = {
    title: 'Quote Title',
    kicker: 'Quote Kicker',
    children: 'Important Quote that is meaningful to the content you want to share.',
    attribution: 'Name of person who originated quote',
    degree: 'Degree',
    image: 'https://showme.missouri.edu/wp-content/uploads/2021/09/091321Truman1-600x600.jpg',
};
