// @university-of-missouri/mizzou-design-system/src/stories/components/Dropdown/Dropdown.stories.tsx
import React from 'react'
import { Story, Meta } from '@storybook/react'

import Dropdown, { DropdownProps } from 'Components/Dropdown/Dropdown'
import { description, status } from 'Components/Dropdown/config.json'
import DropdownHTMLTwig from 'Components/Dropdown/Dropdown.html.twig'
import dropdown from '../../../scripts/dropdown'
import mdx from './Dropdown.mdx'
// import DropdownTwig from './Dropdown.example.twig'

const Template: Story<DropdownProps> = (args, { globals: { fileTypes } }) => {
  React.useEffect(() => {
    dropdown()
  }, [])

  if (fileTypes === 'react') {
    return (
      <Dropdown>
        <Dropdown.Button>I am dropdown button</Dropdown.Button>
        <Dropdown.Content>I am the content.</Dropdown.Content>
      </Dropdown>
    )
  }

  return <div dangerouslySetInnerHTML={{ __html: DropdownHTMLTwig() }}></div>
}

export default {
  component: Dropdown,
  title: 'Design System/Components/Dropdowns',
  parameters: {
    docs: {
      description: {
        component: description,
      },
      page: mdx,
    },
    status,
  },
  argTypes: {
    children: {
      table: {
        category: 'Props',
      },
    },
    attributes: {
      table: {
        disable: true,
      },
    },
    primary: {
      table: {
        category: 'Options',
      },
    },
    secondary: {
      table: {
        category: 'Options',
      },
    },
    branded: {
      table: {
        category: 'Options',
      },
    },
    variant: {
      table: {
        category: 'Options',
      },
    },
    size: {
      table: {
        category: 'Options',
      },
    },
  },
} as Meta

export const dropdowns = Template.bind({})
dropdowns.args = {}

// export const Twig = TwigTemplate.bind({})
