// @university-of-missouri/mizzou-design-system/dist/es5/stories/components/Layers/FeaturedNews/FeaturedNews.stories.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.featuredNews = void 0;
var jsx_runtime_1 = require("react/jsx-runtime");
var FeaturedNews_1 = __importDefault(require("Components/Layers/FeaturedNews/FeaturedNews"));
var config_json_1 = require("Components/Layers/FeaturedNews/config.json");
var FeaturedNews_mdx_1 = __importDefault(require("./FeaturedNews.mdx"));
var Template = function (args) {
    return (0, jsx_runtime_1.jsx)(FeaturedNews_1.default, __assign({}, args));
};
exports.default = {
    component: FeaturedNews_1.default,
    title: 'Design System/Layers/Featured News',
    parameters: {
        docs: {
            description: {
                component: config_json_1.description,
            },
            page: FeaturedNews_mdx_1.default,
        },
        status: config_json_1.status,
    },
    argTypes: {
        title: {
            table: {
                category: 'Options',
            },
        },
        kicker: {
            table: {
                category: 'Options',
            },
        },
        posts: {
            table: {
                category: 'Options',
            },
        },
        buttons: {
            table: {
                category: 'Options',
            },
        },
    },
};
exports.featuredNews = Template.bind({});
exports.featuredNews.args = {
    title: 'Featured News',
    kicker: 'Mizzou Made',
    buttons: [{ title: 'More News', link: 'https://missouri.edu/' }],
    posts: [
        {
            title: 'Mizzou on the moon',
            content: 
            // eslint-disable-next-line prettier/prettier
            '50th anniversary of the moon landing shines spotlight on MU\'s ties to the space program.',
            image: 'https://missouri.edu/sites/default/files/2020-07/CampusBeautyAerialOnJesse.jpg',
            branded: false,
        },
        {
            title: 'Mizzou on the moon',
            content: 
            // eslint-disable-next-line prettier/prettier
            '50th anniversary of the moon landing shines spotlight on MU\'s ties to the space program.',
            image: 'https://missouri.edu/sites/default/files/2020-07/CampusBeautyAerialOnJesse.jpg',
            branded: false,
        },
        {
            title: 'Mizzou on the moon',
            content: 
            // eslint-disable-next-line prettier/prettier
            '50th anniversary of the moon landing shines spotlight on MU\'s ties to the space program.',
            image: 'https://missouri.edu/sites/default/files/2020-07/CampusBeautyAerialOnJesse.jpg',
            branded: false,
        },
    ],
};
