// @university-of-missouri/mizzou-design-system/dist/es5/stories/components/Layers/CallToAction/CallToAction.stories.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.callToAction = void 0;
var jsx_runtime_1 = require("react/jsx-runtime");
var CallToAction_1 = __importDefault(require("Components/Layers/CallToAction/CallToAction"));
var config_json_1 = require("Components/Layers/CallToAction/config.json");
var CallToAction_mdx_1 = __importDefault(require("./CallToAction.mdx"));
var Template = function (args) {
    return (0, jsx_runtime_1.jsx)(CallToAction_1.default, __assign({}, args));
};
exports.default = {
    component: CallToAction_1.default,
    title: 'Design System/Layers/Call To Action',
    parameters: {
        docs: {
            description: {
                component: config_json_1.description,
            },
            page: CallToAction_mdx_1.default,
        },
        status: config_json_1.status,
    },
    argTypes: {
        title: {
            table: {
                category: 'Props',
            },
        },
        children: {
            table: {
                category: 'Props',
            },
        },
        attributes: {
            table: {
                disable: true,
            },
        },
        variant: {
            table: {
                category: 'Options',
            },
        },
        image: {
            table: {
                category: 'Options',
            },
        },
        buttons: {
            table: {
                category: 'Options',
            },
        },
    },
};
exports.callToAction = Template.bind({});
exports.callToAction.args = {
    title: 'Call to Action Title',
    children: 'Call to Action Content',
    variant: 'black',
    image: 'https://missouri.edu/sites/default/files/2020-07/CampusBeautyAerialOnJesse.jpg',
    buttons: [{ title: 'Call to Action Button', link: 'https://missouri.edu/' }],
};
