// @university-of-missouri/mizzou-design-system/src/stories/components/Hero/Hero.stories.tsx
import React from 'react'
import { Story } from '@storybook/react'

import Hero, { HeroProps } from 'Components/Hero/Hero'
import { description, status } from 'Components/Hero/config.json'
import mdx from './Hero.mdx'

const Template: Story<HeroProps> = (args) => {
  return <Hero {...args} />
}

export default {
  component: Hero,
  title: 'Design System/Components/Hero',
  parameters: {
    docs: {
      description: {
        component: description,
      },
      page: mdx,
    },
    status,
  },
  argTypes: {
    attributes: {
      table: {
        disable: true,
      },
    },
    title: {
      table: {
        category: 'Options',
      },
    },
    overline: {
      table: {
        category: 'Options',
      },
    },
    image: {
      table: {
        category: 'Options',
      },
    },
    buttons: {
      table: {
        category: 'Options',
      },
    },
    brand: {
      table: {
        category: 'Options',
      },
    },
    align: {
      table: {
        category: 'Options',
      },
    },
  },
}

export const Default = Template.bind({})
Default.args = {
  title: 'We’ll Show You How.',
  overline: 'Ready to Change the World?',
  image: '',
  buttons: [{ title: 'Apply Today' }, { title: 'Visit Campus' }],
  brand: false,
  align: 'left',
}

export const Brand = Template.bind({})
Brand.args = {
  title: 'We’ll Show You How.',
  overline: 'Ready to Change the World?',
  image: 'https://missouri.edu/sites/default/files/2020-07/CampusBeautyAerialOnJesse.jpg',
  buttons: [{ title: 'Apply Today' }, { title: 'Visit Campus' }],
  brand: true,
  align: 'center',
}
