// @wordpress/interactivity/build/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "createElement", {
  enumerable: true,
  get: function () {
    return _preact.h;
  }
});
Object.defineProperty(exports, "deepSignal", {
  enumerable: true,
  get: function () {
    return _deepsignal.deepSignal;
  }
});
Object.defineProperty(exports, "directive", {
  enumerable: true,
  get: function () {
    return _hooks.directive;
  }
});
Object.defineProperty(exports, "navigate", {
  enumerable: true,
  get: function () {
    return _router.navigate;
  }
});
Object.defineProperty(exports, "prefetch", {
  enumerable: true,
  get: function () {
    return _router.prefetch;
  }
});
Object.defineProperty(exports, "store", {
  enumerable: true,
  get: function () {
    return _store.store;
  }
});
Object.defineProperty(exports, "useContext", {
  enumerable: true,
  get: function () {
    return _hooks2.useContext;
  }
});
Object.defineProperty(exports, "useEffect", {
  enumerable: true,
  get: function () {
    return _hooks2.useEffect;
  }
});
Object.defineProperty(exports, "useMemo", {
  enumerable: true,
  get: function () {
    return _hooks2.useMemo;
  }
});
var _directives = _interopRequireDefault(require("./directives"));
var _router = require("./router");
var _store = require("./store");
var _hooks = require("./hooks");
var _preact = require("preact");
var _hooks2 = require("preact/hooks");
var _deepsignal = require("deepsignal");
/**
 * Internal dependencies
 */

document.addEventListener('DOMContentLoaded', async () => {
  (0, _directives.default)();
  await (0, _router.init)();
  _store.afterLoads.forEach(afterLoad => afterLoad(_store.rawStore));
});
