// @university-of-missouri/mizzou-design-system/src/stories/components/Layers/SplitLayout/SplitLayout.stories.tsx
import React from 'react'
import { Story } from '@storybook/react'

import SplitLayout, { SplitLayoutProps } from 'Components/Layers/SplitLayout/SplitLayout'
import { description, status } from 'Components/Layers/SplitLayout/config.json'
import mdx from './SplitLayout.mdx'

const Template: Story<SplitLayoutProps> = (args) => <SplitLayout {...args} />

export default {
  component: SplitLayout,
  title: 'Design System/Layers/Split Layout',
  parameters: {
    docs: {
      description: {
        component: description,
      },
      page: mdx,
    },
    status,
  },
  argTypes: {
    title: {
      table: {
        category: 'Props',
      },
    },
    kicker: {
      table: {
        category: 'Props',
      },
    },
    children: {
      table: {
        category: 'Props',
      },
    },
    attributes: {
      table: {
        disable: true,
      },
    },
    variant: {
      options: ['default', 'black', 'image', 'offset'],
      table: {
        category: 'Options',
      },
    },
    align: {
      table: {
        category: 'Options',
      },
    },
    image: {
      table: {
        category: 'Options',
      },
    },
    backgroundImage: {
      table: {
        category: 'Options',
      },
    },
    buttons: {
      table: {
        category: 'Options',
      },
    },
  },
}
export const splitLayout = Template.bind({})
splitLayout.args = {
  title: 'Split Layout Title',
  kicker: 'Split Layout Kicker',
  children: <p>Looking for a vibrant community offering endless opportunities to learn and grow? <a href="//missouri.edu/">It’s always a great time to be a Tiger</a>. Schedule a campus tour or try a virtual chat with current students to discover why you’ll love MU.</p>,
  variant: 'black',
  align: 'left',
  image: 'https://missouri.edu/sites/default/files/2020-07/CampusBeautyAerialOnJesse.jpg',
  buttons: [{ title: 'Split Layout Button', link: 'https://missouri.edu/' }],
}
