// @university-of-missouri/mizzou-design-system/.storybook/blocks/KitchenSink/KitchenSink.tsx
import classNames from 'classnames';
import React from 'react';

import './_kitchen-sink.scss'

export interface KitchenSinkProps {
  children: React.ReactNode
}

export const KitchenSink = ({ children }: KitchenSinkProps): JSX.Element => {
  return (
    <div className='mizsb-sink'>{children}</div>
  )
}

export interface KitchenSinkGroupProps {
  children: React.ReactNode
  label?: string
  background?: string
}

export const KitchenSinkGroup = ({ label = '', background = '', children }: KitchenSinkGroupProps): JSX.Element => {

  const sinkClasses = classNames('mizsb-sink__group',
    {
      'mizsb-sink__group--dark': background == 'dark'
    }
  )

  return (
    <div className={sinkClasses}>
      <h2 className='mizsb-sink__group-label'>{label}</h2>
      {children}
    </div>
  )
}


export interface KitchenSinkItemProps {
  label?: string
  children: React.ReactNode
}

export const KitchenSinkItem = ({ label = '', children }: KitchenSinkItemProps): JSX.Element => {
  return (
    <div className='mizsb-sink__item'>
      <KitchenSinkItemLabel text={label} />
      {children}
    </div>
  )
}

export interface KitchenSinkItemLabelProps {
  text: string
}

const KitchenSinkItemLabel = ({ text }: KitchenSinkItemLabelProps): JSX.Element => {
  return (
    <p className="mizsb-sink__item-label">{text}</p>
  )
}
