// @university-of-missouri/eslint-config/rules/base.js
module.exports = {
  rules: {
    // Require or disallow named function expressions
    // https://eslint.org/docs/rules/func-names
    'func-names': ['error', 'as-needed'],

    // Enforce a maximum line length
    // https://eslint.org/docs/rules/max-len
    'max-len': [
      'error',
      {
        code: 100,
        comments: 80,
        ignoreTrailingComments: true,
        ignoreUrls: true,
      },
    ],

    // Disallow use of console
    // https://eslint.org/docs/latest/rules/no-console
    'no-console': ['warn', { allow: ['error', 'debug', 'group', 'groupEnd', 'info', 'warn'] }],

    // Disallow dangling underscores in identifiers
    // https://eslint.org/docs/latest/rules/no-underscore-dangle
    'no-underscore-dangle': 'off',

    // Enforce the consistent use of either backticks, double, or single quotes
    // https://eslint.org/docs/latest/rules/quotes
    quotes: [
      'error',
      'single',
      { avoidEscape: true, allowTemplateLiterals: false },
    ],

    // Require or disallow semicolons instead of ASI
    // https://eslint.org/docs/latest/rules/semi
    semi: ['error', 'never'],
  },
}
