// @university-of-missouri/mizzou-design-system/dist/es5/stories/components/Colophon/Colophon.stories.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.colophon = void 0;
var jsx_runtime_1 = require("react/jsx-runtime");
var Colophon_1 = __importDefault(require("Components/Colophon/Colophon"));
var config_json_1 = require("Components/Colophon/config.json");
var Colophon_mdx_1 = __importDefault(require("./Colophon.mdx"));
var Template = function (args) {
    return (0, jsx_runtime_1.jsx)(Colophon_1.default, __assign({}, args));
};
exports.default = {
    component: Colophon_1.default,
    title: 'Design System/Components/Colophon',
    parameters: {
        docs: {
            description: {
                component: config_json_1.description,
            },
            page: Colophon_mdx_1.default,
        },
        status: config_json_1.status,
    },
    argTypes: {
        attributes: {
            table: {
                disable: true,
            },
        },
        department: {
            table: {
                category: 'Options',
            },
        },
    },
};
exports.colophon = Template.bind({});
exports.colophon.args = {
    department: [
        {
            name: 'Digital Service',
            address: '234 Heinkel Building, Columbia, MO 65211',
            email: 'digitalservice@missouri.edu',
        },
    ],
};
