// @university-of-missouri/mizzou-design-system/dist/es5/stories/layout/Flex.stories.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Flex = void 0;
var jsx_runtime_1 = require("react/jsx-runtime");
var classnames_1 = __importDefault(require("classnames"));
var Layout_1 = require("Layout/Layout");
var TestComponent = function (_a) {
    var className = _a.className;
    var testComponentClass = (0, classnames_1.default)(className);
    return ((0, jsx_runtime_1.jsxs)("div", __assign({ className: testComponentClass }, { children: [(0, jsx_runtime_1.jsx)("div", __assign({ className: 'miz-fill--black-100', style: { padding: '1rem', borderRadius: '8px' } }, { children: "Panel 1" })), (0, jsx_runtime_1.jsx)("div", __assign({ className: 'miz-fill--black-100', style: { padding: '1rem', borderRadius: '8px' } }, { children: "Panel 2" }))] })));
};
var Template = (0, Layout_1.addLayout)(function (_a) {
    var args = __rest(_a, []);
    return ((0, jsx_runtime_1.jsx)(TestComponent, __assign({}, args)));
});
exports.default = {
    title: 'Design System/Layout',
    component: TestComponent,
    args: {
        allowLayout: true,
        advanced: false,
    },
    argTypes: {
        allowLayout: {
            name: 'Allow layouts',
            table: {
                category: 'Layout',
                disable: true,
            },
        },
        layout: {
            name: 'Layout',
            table: {
                category: 'Layout',
            },
        },
        advanced: {
            name: 'Toggle advanced options',
            control: 'boolean',
            table: {
                category: 'Advanced',
            },
        },
        className: {
            name: 'Add CSS',
            type: 'string',
            table: {
                category: 'Advanced',
            },
            if: {
                arg: 'advanced',
            },
        },
    },
    decorators: [
        function (Story) { return ((0, jsx_runtime_1.jsx)("div", __assign({ className: 'miz-container' }, { children: (0, jsx_runtime_1.jsx)(Story, {}) }))); },
    ],
};
exports.Flex = Template.bind({});
exports.Flex.args = {
    allowLayout: true,
    layout: { type: 'flex', direction: 'row', gap: true },
};
