// @university-of-missouri/mizzou-design-system/src/stories/components/Navigation/Breadcrumbs.stories.tsx
import React from 'react'
import { Story, Meta } from '@storybook/react'

import Breadcrumbs, { BreadcrumbsProps } from 'Components/Breadcrumbs/Breadcrumbs'
import { description, status } from 'Components/Breadcrumbs/config.json'
// import TwigBreadcrumbs from './Breadcrumbs.twig'
import mdx from './Breadcrumbs/Breadcrumbs.mdx'

const Template: Story<BreadcrumbsProps> = (args: any) => {
  return <Breadcrumbs {...args} />
}

export default {
  component: Breadcrumbs,
  title: 'Design System/Components/Navigation/Breadcrumbs',
  decorators: [
    (BreadcrumbsStory) => (
      <div className="miz-main-grid miz-fill--black-100">
        <BreadcrumbsStory />
      </div>
    ),
  ],
  parameters: {
    docs: {
      description: {
        component: description,
      },
      page: mdx,
    },
    status,
  },
  argTypes: {
    children: {
      table: {
        category: 'Props',
      },
    },
    attributes: {
      table: {
        disable: true,
      },
    },
    branded: {
      table: {
        category: 'Options',
      },
    },
  },
} as Meta

export const breadcrumbs: Story = Template.bind({})
breadcrumbs.args = {
  branded: false,
  list: {
    items: [
      {
        link: {
          text: 'Home',
          href: '/',
        },
      },
      {
        link: {
          text: 'About us',
          href: '/crumb-1/',
        },
      },
      {
        link: {
          text: 'About us',
          href: '/crumb-1/',
        },
      },
      {
        page: {
          text: 'Page title',
        },
      },
    ],
  },
}
