// @university-of-missouri/mizzou-design-system/src/stories/components/Navigation/MegaMenu.stories.tsx
import React from 'react'
import { Story } from '@storybook/react'

import MegaMenu, { MegaMenuProps } from 'Components/MegaMenu/MegaMenu'
import MegaMenuTwig from 'Components/MegaMenu/MegaMenu.twig'
import { description, status } from 'Components/MegaMenu/config.json'
import Nav from 'Components/Nav/Nav'
import dropdown from '../../../scripts/dropdown'
import mdx from './MegaMenu/MegaMenu.mdx'
import megaMenuContext from './MegaMenu/MegaMenu.context'

const Template: Story<MegaMenuProps> = (args, { globals: { fileTypes } }) => {
  React.useEffect(() => {
    dropdown()
  }, [])

  if (fileTypes === 'react') {
    return <MegaMenu {...args} />
  }

  return <div dangerouslySetInnerHTML={{ __html: MegaMenuTwig() }}></div>
}

export default {
  component: MegaMenu,
  title: 'Design System/Components/Navigation/Mega Menu',
  parameters: {
    docs: {
      description: {
        component: description,
      },
      page: mdx,
    },
    status,
  },
  argTypes: {},
  subcomponents: { Nav },
}

export const megaMenu = Template.bind({})
megaMenu.args = {
  ...megaMenuContext.default,
}
