// @university-of-missouri/mizzou-design-system/src/stories/components/Navigation/Primary.stories.tsx
import React from 'react'
import { Story } from '@storybook/react'

import Nav from 'Components/Nav/Nav'
// import primary from 'Components/Navigation/Primary/primary'
import PrimaryNavigation, {
  PrimaryNavigationProps,
} from 'Components/Navigation/Primary/PrimaryNavigation'
import { description, status } from 'Components/Navigation/Primary/config.json';
// import expand from '../../../scripts/expand'
import mdx from './Primary/Primary.mdx'
import context from './Primary/Primary.context'
// import PrimaryNavigationTwig from './Primary/Primary.example.twig'

const Template: Story<PrimaryNavigationProps> = (args) => {
  return <PrimaryNavigation {...args} />
}

// const TwigTemplate: Story = () => {
//   React.useEffect(() => {
//     primary()
//     expand()
//   }, [])

//   return <div dangerouslySetInnerHTML={{ __html: PrimaryNavigationTwig({}) }}></div>
// }

export default {
  component: PrimaryNavigation,
  subcomponents: { Nav },
  title: 'Design System/Components/Navigation/Primary Navigation',
  parameters: {
    docs: {
      description: {
        component: description,
      },
      page: mdx,
    },
    status,
  },
  argTypes: {
    attributes: {
      table: {
        disable: true,
      },
    },
  },
}

export const primaryNavigation = Template.bind({})
primaryNavigation.args = {
  ...context.default.list,
}

// export const Twig = TwigTemplate.bind({})
// Twig.args = {}
