// @university-of-missouri/mizzou-design-system/src/components/Header/Header.tsx
import * as React from 'react'
import DrupalAttribute from 'drupal-attribute'
import classNames from 'classnames'

import Ribbon from 'Components/Ribbon/Ribbon'
import Masthead from 'Components/Masthead/Masthead'
// import PrimaryNavigation from 'Components/Navigation/Primary/PrimaryNavigation'

export interface HeaderProps {
  attributes: DrupalAttribute
  theme: 'light' | 'dark'
  ribbonContainer: React.ReactNode
  mastheadContainer: React.ReactNode
  primaryNavContainer: React.ReactNode
}

const Header = ({
  attributes = new DrupalAttribute(''),
  theme = 'light',
  ribbonContainer = <Ribbon />,
  mastheadContainer = <Masthead />,
  primaryNavContainer,
}: HeaderProps): JSX.Element => {
  const HeaderAttributes = {
    className: classNames(
      'miz-header',
      {
        'miz-header--dark': theme === 'dark',
      },
      attributes.get('class'),
    ),
  }

  attributes.forEach((value, key) => {
    if (key !== 'class') {
      HeaderAttributes[key] = value
    }
  })

  return (
    <header {...HeaderAttributes}>
      <div className="miz-fill--black">
        <div className="miz-container">{ribbonContainer}</div>
      </div>

      <div className={theme === 'light' ? 'miz-fill--white' : 'miz-fill--black-600'}>
        <div className="miz-container">{mastheadContainer}</div>
      </div>

      <div className="miz-fill--gold">
        <div className="miz-container">{primaryNavContainer}</div>
      </div>
    </header>
  )
}

export default Header
