// @university-of-missouri/mizzou-design-system/src/components/InputGroup/Search/Search.tsx
import * as React from 'react'
import DrupalAttribute from 'drupal-attribute'
import classNames from 'classnames'

import Button from 'Components/Button/Button'

export interface SearchFormProps {
  attributes?: DrupalAttribute
  formAction?: string
  labelText?: string
  inputId?: string
  inputName?: string
  inputPlaceholder?: string
}

const SearchForm = ({
  attributes = new DrupalAttribute(''),
  formAction = '/search/',
  labelText = 'Search',
  inputId = 'search-input',
  inputName = 'q',
  inputPlaceholder = 'Search this site',
}: SearchFormProps): JSX.Element => {
  const SearchFormAttributes = {
    className: classNames('miz-input-group', attributes.get('class')),
  }

  attributes.forEach((value, key) => {
    if (key !== 'class') {
      SearchFormAttributes[key] = value
    }
  })

  return (
    <form action={formAction} method="get" {...SearchFormAttributes}>
      <label htmlFor={inputId} className="miz-label--hidden">
        {labelText}
      </label>
      <input
        type="text"
        className="miz-input"
        id={inputId}
        placeholder={inputPlaceholder}
        name={inputName}
      />
      <Button
        primary={true}
        size={'small'}
        attributes={attributes.addClass(
          'miz-button--icon',
          'miz-button--square-sm',
          'miz-input-group__button',
        )}
      >
        <i className="miz-icon material-icons miz-icon--button miz-icon--md">search</i>
      </Button>
    </form>
  )
}

export default SearchForm
