// @university-of-missouri/mizzou-design-system/src/stories/components/BlockMessage/BlockMessage.stories.tsx
import React from 'react'
import { Story } from '@storybook/react'

import BlockMessage, { BlockMessageProps } from 'Components/BlockMessage/BlockMessage'
import { description, status } from 'Components/BlockMessage/config.json'
import mdx from './BlockMessage.mdx'

const Template: Story<BlockMessageProps> = (args) => {
  return <BlockMessage {...args} />
}

export default {
  component: BlockMessage,
  title: 'Design System/Components/Block Message',
  parameters: {
    docs: {
      description: {
        component: description,
      },
      page: mdx,
    },
    status,
  },
  argTypes: {
    title: {
      table: {
        category: 'Props',
      },
    },
    children: {
      table: {
        category: 'Props',
      },
    },
    attributes: {
      table: {
        disable: true,
      },
    },
    variant: {
      table: {
        category: 'Options',
      },
    },
  },
}
export const Base = Template.bind({})
Base.args = {
  title: 'Block Message',
  children: 'Message Content',
}

export const ThemedVariants = Template.bind({})
ThemedVariants.args = {
  title: 'Block Message Message',
  children: 'Block Message Content',
  variant: 'info',
}
