// @university-of-missouri/mizzou-design-system/src/stories/components/Layers/Quote/Quote.stories.tsx
import React from 'react'
import { Story } from '@storybook/react'

import Quote, { QuoteProps } from 'Components/Layers/Quote/Quote'
import { description, status } from 'Components/Layers/Quote/config.json'
import mdx from './Quote.mdx'

const Template: Story<QuoteProps> = (args) => {
  return <Quote {...args} />
}

export default {
  component: Quote,
  title: 'Design System/Layers/Quote',
  parameters: {
    docs: {
      description: {
        component: description,
      },
      page: mdx,
    },
    status,
  },
  argTypes: {
    title: {
      table: {
        category: 'Props',
      },
    },
    kicker: {
      table: {
        category: 'Props',
      },
    },
    attributes: {
      table: {
        disable: true,
      },
    },
    attribution: {
      table: {
        category: 'Options',
      },
    },
    degree: {
      table: {
        category: 'Options',
      },
    },
    children: {
      table: {
        category: 'Options',
      },
    },
    image: {
      table: {
        category: 'Options',
      },
    },
    buttons: {
      table: {
        category: 'Options',
      },
    },
  },
}
export const quote = Template.bind({})
quote.args = {
  title: 'Quote Title',
  kicker: 'Quote Kicker',
  children: 'Important Quote that is meaningful to the content you want to share.',
  attribution: 'Name of person who originated quote',
  degree: 'Degree',
  image: 'https://showme.missouri.edu/wp-content/uploads/2021/09/091321Truman1-600x600.jpg',
}
